<div class="container">

<table style="width: 100%;"><tr>
<td>QC_plots</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>QQ and Manhattan plots</h2>

<h3>Description</h3>

<p>This function creates the most important graphs of the QC: the
QQ plots and the Manhattan plot. It also calculates lambda,
and determines the effect of the filters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">QC_plots(dataset,
         plot_QQ = TRUE, plot_Man = TRUE,
         FRQfilter_values = NULL, FRQfilter_NA = filter_NA,
         HWEfilter_values = NULL, HWEfilter_NA = filter_NA,
         calfilter_values = NULL, calfilter_NA = filter_NA,
         impfilter_values = NULL, impfilter_NA = filter_NA,
         impfilter_min = min(dataset$IMP_QUALITY, na.rm = TRUE),
         manfilter_FRQ = NULL, manfilter_HWE = NULL,
         manfilter_cal = NULL, manfilter_imp = NULL,
         filter_NA = TRUE,
         plot_cutoff_p = 0.05, plot_names = FALSE,
         QQ_colors = c("red", "blue", "orange", "green3", "yellow"),
         plot_QQ_bands = FALSE,
         save_name = "dataset", save_dir = getwd(),
         header_translations, use_log = FALSE,
         check_impstatus = FALSE, ignore_impstatus = FALSE,
         T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
         F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
         NA_strings = c(NA, "NA", ".", "-"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataset</code></td>
<td>
<p>vector of p-values or a data frame containing
the p-value column and (depending on the settings) columns
for chromosome number, position, the quality parameters,
sample size and imputation status.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_QQ, plot_Man</code></td>
<td>
<p>logical; should QQ and Manhattan plots
be saved?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FRQfilter_values, HWEfilter_values,
calfilter_values, impfilter_values</code></td>
<td>

<p>numeric vectors; the threshold values for the QQ plot filters.
The filters are for allele-frequency, HWE p-values, callrate
and imputation-quality parameters, respectively. A maximum of
five values can be specified per parameter.
</p>

<ul>
<li>
<p> Set to <code>NULL</code> to disable the QQ filter for that
parameter.
</p>
</li>
<li>
<p> To filter missing values only, set the filter value
to <code>NA</code> and the corresponding <code>filter_NA</code>
argument to <code>TRUE</code>.
</p>
</li>
<li>
<p> The allele-frequency filter is two-sided: for a
filter-value of <code>x</code>, it will exclude entries
with <code>freq &lt; x</code> and <code>freq &gt; 1 - x</code>.
</p>
</li>
<li>
<p> Values &gt;= 1 will be divided by the SNP's sample size.
This allows sample-size dependent filtering of allele
frequencies. Note that this uses the sample size reported
in the sample-size column for that specific SNP. SNPs
without sample size will be excluded if the corresponding
<code>filter_NA</code> argument is <code>TRUE</code> and ignored if
it is <code>FALSE</code>.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FRQfilter_NA, HWEfilter_NA, calfilter_NA,
impfilter_NA, filter_NA</code></td>
<td>

<p>logical; should the filters exclude (<code>TRUE</code>) or
ignore (<code>FALSE</code>) missing values? <code>filter_NA</code>
is the default setting, the others allow
variable-specific settings.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impfilter_min</code></td>
<td>
<p>numeric; the lowest possible value for
imputation-quality. This argument is currently redundant, as
it is calculated automatically.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>manfilter_FRQ, manfilter_HWE, manfilter_cal, manfilter_imp</code></td>
<td>

<p>single, numeric values; the filter-settings for allele-frequency,
HWE p-values, callrate and imputation quality respectively,
for the Manhattan plot. The arguments are passed to
<code>HQ_filter</code>.
To filter missing values only, set to <code>NA</code> and the
corresponding <code>filter_NA</code> argument to <code>TRUE</code>. To
disable filtering entirely, set to <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_cutoff_p</code></td>
<td>
<p>numeric; the threshold of p-values to be
shown in the QQ &amp; Manhattan plots. Higher (less
significant) p-values are excluded from the plot. The default
setting is <code>0.05</code>, which excludes 95% of data-points.
It's not recommended to increase the value above <code>0.05</code>,
as this may dramatically increase running time and memory usage.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_names</code></td>
<td>
<p>argument currently redundant.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QQ_colors</code></td>
<td>
<p>vector of R color-values; the color of the
QQ filter-plots. The unfiltered data is black by default.
This argument sets the colors of the least (first value) to
most (last value) stringent filters. (For this setting,
filter values <code>&gt;= 1</code> (i.e. sample-size based filtering)
are considered less stringent than values <code>&lt; 1</code>.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_QQ_bands</code></td>
<td>
<p>logical; should probability bands be
added to the QQ plot?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_name</code></td>
<td>
<p>character string; the filename, <em>without</em>
extension, for the graphs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_dir</code></td>
<td>
<p>character string; the directory where the
graphs are saved. Note that R uses <em>forward</em>
slash (/) where Windows uses the backslash (\).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header_translations</code></td>
<td>
<p>translation table for column names.
See <code>translate_header</code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names of <code>QC_GWAS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_log</code></td>
<td>
<p>argument used by <code>QC_GWAS</code>;
redundant when <code>QC_plots</code> is used separately.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_impstatus</code></td>
<td>
<p>logical; should the imputation-status
column be passed to <code>convert_impstatus</code>?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_impstatus</code></td>
<td>
<p>logical; if <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T_strings, F_strings, NA_strings</code></td>
<td>
<p>arguments passed
to <code>convert_impstatus</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function <code>QC_plots</code> grew out of phase 4 of
<code>QC_GWAS</code>. It carries out three functions, hence
the vague name: it calculates  lambda, it applies the
QQ filters, and it creates the QQ and Manhattan plots (a
separate function is available for creating
regional-association plots: see below). The function schematic
is as follows:
</p>

<ul>
<li>
<p> Preparing the dataset: this step involves translating
the dataset header to the standard column-names (by
<code>identify_column</code>) and converting imputation
status (by <code>convert_impstatus</code>). Both steps
are optional, and are disabled by default. If the function
cannot identify the imputation status column, it will
generate a warning message and disable the
imputation-status dependent filters.
</p>
</li>
<li>
<p> Calculating the QC stats: here it generates the
filters an calculates how many SNPs are removed. Lambda
is also calculated at this point.
</p>
</li>
<li>
<p> Creating a QQ graph of every variable for which
filters have been specified. Every graph contains an
unfiltered plot, plus plots for every effective filter.
("Effective" means "excludes more SNPs than the
previous, less-stringent filter".)
</p>
</li>
<li>
<p> Creating the Manhattan plot. The default Manhattan
plot covers chromosomes 1 to 23 (X). Fields for XY, Y and
M are added when such SNPs are present.
</p>
</li>
</ul>
<h3>Value</h3>

<p>An object of class 'list' with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>vector of the lambda values of all SNPs,
genotyped SNPs and imputed SNPs, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_impstatus</code></td>
<td>
<p>logical value indicating whether
imputation status was used when applying the filters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FRQfilter_names, HWEfilter_names, calfilter_names, impfilter_names</code></td>
<td>

<p>character vectors naming the specified QQ filters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FRQfilter_N, HWEfilter_N, calfilter_N, impfilter_N</code></td>
<td>

<p>numeric vectors; the number of SNPs removed by the specified
filters. Note that the filters are sorted before being
applied, so the order may not match that of the input.
Check the <code>filter_names</code> output to see the order
that was used inside <code>QC_plots</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Manfilter_N</code></td>
<td>
<p>numeric; the number of SNPs removed by the
Manhattan filter. This does not include those SNPs removed
because they lacked p or chromosome/position-values, or failed
the p-cutoff threshold.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>By default, <code>QC_plots</code> expects <code>dataset</code> to use the
standard column-names used by <code>QC_GWAS</code>. A
translation table can be specified in <code>header_translations</code>
to allow non-standard names. See <code>translate_header</code>
for more information.
</p>
<p>The function accepts both integer and character chromosome
values. Character values of <code>"X"</code>, <code>"Y"</code>, <code>"XY"</code>
and <code>"M"</code> are automatically converted to integers. By
default, the Manhattan plot shows all autosomal
chromosomes and chromosome X. Fields for Y, XY and M are
added only when such SNPs are present.
</p>
<p>There must be more than 10 p-values at or below the
<code>plot_cutoff_p</code> threshold for the QQ and Manhattan
plots to be created.  
</p>


<h3>See Also</h3>

<p><code>plot_regional</code> for creating a regional association
plot.
</p>
<p><code>check_P</code> for comparing the reported p-values to
the p expected from the effect size and standard error.
</p>
<p><code>QQ_plot</code> for generating simpler QQ plots.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ## Not run: 
  data("gwa_sample")

  QC_plots(dataset = gwa_sample,
    plot_QQ = TRUE, plot_QQ_bands = TRUE, plot_Man = TRUE,
    FRQfilter_values = c(NA, 0.01, 0.05, 3),
    calfilter_values = c(NA, 0.95, 0.99),
    manfilter_FRQ = 0.05, manfilter_cal = 0.95,
    filter_NA = TRUE, save_name = "sample_plots")
    
## End(Not run)
</code></pre>


</div>