<div class="container">

<table style="width: 100%;"><tr>
<td>QQ_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>QQ plot(s) of expected vs. reported p-values</h2>

<h3>Description</h3>

<p><code>QQ_plot</code> generates a simple QQ plot of the expected and
reported p-value distribution. It includes the option to
filter the data with the high-quality filter. <code>QQ_series</code>
generates a series of such QQ plots for multiple filter
settings.
</p>


<h3>Usage</h3>

<pre><code class="language-R">QQ_plot(dataset, save_name = "dataset", save_dir = getwd(),
        filter_FRQ = NULL, filter_cal = NULL,
        filter_HWE = NULL, filter_imp = NULL,
        filter_NA = TRUE,
        filter_NA_FRQ = filter_NA, filter_NA_cal = filter_NA,
        filter_NA_HWE = filter_NA, filter_NA_imp = filter_NA,
        p_cutoff = 0.05, plot_QQ_bands = FALSE,
        header_translations,
        check_impstatus = FALSE, ignore_impstatus = FALSE,
        T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
        F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
        NA_strings = c(NA, "NA", ".", "-"), ...)
QQ_series(dataset, save_name = "dataset", save_dir = getwd(),
          filter_FRQ = NULL, filter_cal = NULL,
          filter_HWE = NULL, filter_imp = NULL,
          filter_NA = TRUE,
          filter_NA_FRQ = filter_NA, filter_NA_cal = filter_NA,
          filter_NA_HWE = filter_NA, filter_NA_imp = filter_NA,
          p_cutoff = 0.05, plot_QQ_bands = FALSE,
          header_translations,
          check_impstatus = FALSE, ignore_impstatus = FALSE,
          T_strings = c("1", "TRUE", "yes", "YES", "y", "Y"),
          F_strings = c("0", "FALSE", "no", "NO", "n", "N"),
          NA_strings = c(NA, "NA", ".", "-"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataset</code></td>
<td>
<p>a data frame containing
the p-value column and (depending on the settings) columns
for chromosome number, position, the quality parameters,
sample size and imputation status.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_name</code></td>
<td>
<p>for <code>QQ_plot</code>, a character string;
for <code>QQ_series</code>, a vector of character strings;
specifying the filename(s) of the graph, <em>without</em>
extension.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_dir</code></td>
<td>
<p>character string; the directory where the
output files are saved. Note that R uses <em>forward</em>
slash (/) where Windows uses the backslash (\).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_FRQ, filter_cal, filter_HWE, filter_imp</code></td>
<td>

<p>Filter threshold-values for allele-frequency, callrate,
HWE p-value and imputation quality, respectively. Passed to
<code>HQ_filter</code>. <code>QQ_plot</code> takes only
single values, but  <code>QQ_series</code> accepts vectors
as well (see 'details').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_NA</code></td>
<td>
<p>logical; if <code>TRUE</code>, then missing filter
variables will be excluded; if <code>FALSE</code>, they will be
ignored. <code>QQ_plot</code> takes only single values, but 
<code>QQ_series</code> accepts vectors as well (see
'Details'). <code>filter_NA</code> is the default setting for all
variables; variable-specific settings can be specified with
the following arguments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_NA_FRQ, filter_NA_cal,
filter_NA_HWE, filter_NA_imp</code></td>
<td>

<p>logical; variable-specific settings for <code>filter_NA</code>.
These arguments are passed to <code>HQ_filter</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_cutoff</code></td>
<td>
<p>numeric; the threshold of p-values to be
shown in the QQ plot(s). Higher (less significant) p-values
are excluded from the plot. The default setting is <code>0.05</code>,
which excludes 95% of data-points. It's not recommended to
increase the value above <code>0.05</code>, as this may
dramatically increase running time and memory usage.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_QQ_bands</code></td>
<td>
<p>logical; should probability bands be
added to the QQ plot?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header_translations</code></td>
<td>
<p>translation table for column names.
See <code>translate_header</code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names of <code>QC_GWAS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_impstatus</code></td>
<td>
<p>logical; should the imputation-status
column be passed to <code>convert_impstatus</code>?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_impstatus</code></td>
<td>
<p>logical; if <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T_strings, F_strings, NA_strings</code></td>
<td>
<p>arguments passed
to <code>convert_impstatus</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments passed to <code>plot</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>QQ_series</code> accepts multiple filter-values, and
passes these one by one to <code>QQ_plot</code> to generate a
series of plots. For example, specifying:
</p>
<p><code>filter_FRQ = c(0.05, 0.10), filter_cal = c(0.90, 0.95)</code>
</p>
<p>will generate two plots. The first excludes SNPs with
allele frequency &lt; 0.05 or callrate &lt; 0.90; the second allele 
frequency &lt; 0.10 or callrate &lt; 0.95. The same principle
applies to the <code>NA_filter</code> settings. If the vectors
submitted to the filter arguments are of unequal length, the
shorter vector will be recycled until it equals the length of
the longer (if possible). To filter missing values only, set
the filter to <code>NA</code> and the corresponding NA-filter
argument to <code>TRUE</code>. Setting the filter argument to
<code>NULL</code> will disable the filter entirely, regardless of
the NA-filter setting.
</p>


<h3>Value</h3>

<p>Both functions return an invisible value <code>NULL</code>.</p>


<h3>See Also</h3>

<p><code>QC_plots</code> for generating more complex QQ plots
as well as Manhattan plots.
</p>
<p><code>QC_histogram</code> for creating histograms.
</p>
<p><code>check_P</code> for comparing the reported p-values to
the p expected from the effect size and standard error.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ## Not run: 
    data("gwa_sample")
  
    QQ_plot(dataset = gwa_sample,
            save_name = "sample_QQ",
            filter_FRQ = 0.01, filter_cal = 0.95,
            filter_NA = FALSE)
  
    QQ_series(dataset = gwa_sample,
              save_name = "sample_QQ",
              filter_FRQ = c(NA, 0.01, 0.01),
              filter_cal = c(NA, 0.95, 0.95),
              filter_NA = c(FALSE, FALSE, TRUE))
  
## End(Not run)
</code></pre>


</div>