<div class="container">

<table style="width: 100%;"><tr>
<td>filter_GWAS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Automated filtering and reformatting of GWAS results files</h2>

<h3>Description</h3>

<p>This function was created as a convenient way to automate the
removal of low-quality and non-autosomal SNPs. It
includes the same formatting options as <code>QC_GWAS</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter_GWAS(ini_file,
            GWAS_files, output_names,
            gzip_output = TRUE,
            dir_GWAS = getwd(), dir_output = dir_GWAS,
            FRQ_HQ = NULL, HWE_HQ = NULL,
            cal_HQ = NULL, imp_HQ = NULL,
            FRQ_NA = TRUE, HWE_NA = TRUE,
            cal_NA = TRUE, imp_NA = TRUE,
            ignore_impstatus = FALSE,
            remove_X = FALSE, remove_Y = FALSE,
            remove_XY = FALSE, remove_M = FALSE,
            header_translations,
            check_impstatus = FALSE,
            imputed_T = c("1", "TRUE", "yes", "YES", "y", "Y"),
            imputed_F = c("0", "FALSE", "no", "NO", "n", "N"),
            imputed_NA = NULL,
            column_separators = c("\t", " ", "", ",", ";"),
            header = TRUE, nrows = -1, nrows_test = 1000,
            comment.char = "", na.strings = c("NA", "."),
            out_header = "original", out_quote = FALSE,
            out_sep = "\t", out_eol = "\n", out_na = "NA",
            out_dec = ".", out_qmethod = "escape",
            out_rownames = FALSE, out_colnames = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ini_file</code></td>
<td>

<p>(the filename of) a table listing the files to be
processed and the filters to be applied. See 'Details'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GWAS_files</code></td>
<td>
<p>character vector: when no <code>ini_file</code> is
provided, this identifies the files to be processed. See
'Details'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_names</code></td>
<td>
<p>character vector: the filenames for the
output files. The default option is to use the input
filenames. Note that, unlike with other <code>QCGWAS</code>
functions, the file extensions should be included (However,
the function will automatically add <code>".gz"</code> when the
files are compressed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gzip_output</code></td>
<td>
<p>logical; should the output files be
compressed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir_GWAS, dir_output</code></td>
<td>
<p>character-strings specifying the
directory address of the folders for the input files and the
output, respectively. Note that R uses <em>forward</em> slash
(/) where Windows uses backslash (\).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FRQ_HQ, HWE_HQ, cal_HQ, imp_HQ</code></td>
<td>

<p>Numeric vectors. When no <code>ini_file</code> is provided, these
arguments specify the filter threshold-values for allele
frequency, HWE p-value, callrate and imputation quality,
respectively. Passed to <code>HQ_filter</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FRQ_NA, HWE_NA, cal_NA, imp_NA</code></td>
<td>

<p>Logical vectors. When no <code>ini_file</code> is provided, these
arguments specify whether missing values (of allele frequency,
HWE p-value, callrates and imputation quality, respectively)
are excluded (<code>TRUE</code>) or ignored (<code>FALSE</code>).
Passed to <code>HQ_filter</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_impstatus</code></td>
<td>

<p>Logical vector. When no <code>ini_file</code> is provided, this
argument specifies whether imputation status is taken into
account when applying the filters. If <code>FALSE</code>, HWE p-value
and callrate filters are applied only to genotyped SNPs, and
imputation quality filters only to imputed SNPs. If
<code>TRUE</code>, the filters are applied to all SNPs regardless
of the imputation status.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_X, remove_Y, remove_XY, remove_M</code></td>
<td>

<p>logical; respectively whether X-chromosome, Y-chromosome,
pseudo-autosomal and mitochondrial SNPs are removed. Note:
these arguments accept only a single <code>TRUE</code> or
<code>FALSE</code> value.
Unlike the above settings, it's not possible to specify
them independently for every dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header_translations</code></td>
<td>
<p>translation table for column names.
See <code>translate_header</code> for more information. If
the argument is left empty, <code>dataset</code> is assumed to use
the standard column-names used by <code>QC_GWAS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_impstatus</code></td>
<td>
<p>logical; should
<code>convert_impstatus</code> be called to convert the
imputation-status column into standard values?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputed_T, imputed_F, imputed_NA</code></td>
<td>
<p>arguments passed to
<code>convert_impstatus</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>column_separators</code></td>
<td>
<p>character string or vector; specifies
the values used as column delimitator in the GWAS file(s). The
argument is passed to <code>load_test</code>; see the
description of that function for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nrows_test</code></td>
<td>
<p>integer; the number of rows used for
"trial-loading". Before loading the entire dataset, the
function <code>load_test</code> is called to determine the
dataset's file-format by reading the top <code>x</code> lines, where
<code>x</code> is <code>nrows_test</code>. Setting <code>nrows_test</code> to
a low number
(e.g. <code>150</code>) means quick testing, but runs the risk of
missing problems in lower rows. To test the entire dataset,
set it to <code>-1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header, nrows, comment.char, na.strings, ...</code></td>
<td>

<p>arguments passed to <code>read.table</code> when importing
the dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out_header</code></td>
<td>
<p>Translation table for the column names of
the <em>output</em> file. This argument is the opposite of
<code>header_translations</code>: it translates the standard
column-names of <code>QC_GWAS</code> to user-defined ones.
<code>output_header</code> can be one of three things:
</p>

<ul>
<li>
<p> A user specified table similar to the one used by
<code>translate_header</code>. However, as this
translates standard names into non-standard ones, the
standard names should be in the right column, and the
desired ones in the left. There is also no requirement
for the names in the <em>left</em> column to be uppercase.
</p>
</li>
<li>
<p> The name of a file in <code>dir_GWAS</code> containing
such a table.
</p>
</li>
<li>
<p> Character string specifying a standard form. See
<code>QC_GWAS</code>, section 'Output header' for the
options.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out_quote, out_sep, out_eol, out_na, out_dec,
out_qmethod, out_rownames, out_colnames</code></td>
<td>

<p>arguments passed to
<code>write.table</code> when saving the final dataset.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The easiest way to use <code>filter_GWAS</code> is by passing an ini
file to the <code>ini_file</code> argument.
The ini file can be generated by running <code>QC_series</code>
with the <code>save_filtersettings</code> argument set to <code>TRUE</code>.
The output will include a file 'Check_filtersettings.txt',
describing the (high-quality) filter settings used for each
file (taking into account whether there was enough data, i.e.
whether the <code>use_threshold</code> was met, to apply the filters).
</p>
<p>The <code>ini_file</code> argument accepts both a table
or the name of a file in <code>dir_GWAS</code> or the
current R working directory.
</p>
<p>If no <code>ini_file</code> is specified, the function will use the
<code>GWAS_files</code>, x_HQ, x_NA and <code>ignore_impstatus</code>
arguments to construct such a table.
<code>GWAS_files</code> can either be a character vector or a single
value. If a single string, all filenames containing the string
will be processed. The other arguments can also be a vector or
a single value; if the latter, they will be recycled to create
a vector of the correct length.  
</p>
<p>If neither <code>ini_file</code> nor <code>GWAS_files</code> are specified,
the function will look for a file
<code>Check_filtersettings.txt</code>
in <code>dir_GWAS</code> and the current R working directory.
</p>
<p>Note that <code>ini_file</code> overrules the other filter settings,
i.e. one cannot adjust <code>ini_file</code> through the other
arguments.
</p>


<h3>Value</h3>

<p>An invisible logical vector, indicating which files were
successfully filtered.</p>


<h3>Note</h3>

<p>R is not the optimal platform for filtering GWAS files. This
function was added at the request of a user, but an UNIX script
is likely to be faster.</p>


</div>