<div class="container">

<table style="width: 100%;"><tr>
<td>gwa_sample</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Sample dataset for the QCGWAS package
</h2>

<h3>Description</h3>

<p>A fake GWAS results dataset for use in the examples.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(gwa_sample)</code></pre>


<h3>Format</h3>

<p>A data frame with 10000 observations on the following 15 variables.
</p>

<dl>
<dt><code>MARKER</code></dt>
<dd>
<p>a character vector; the SNP IDs.</p>
</dd>
<dt><code>STRAND</code></dt>
<dd>
<p>a character vector; the DNA strand of
the listed alleles.</p>
</dd>
<dt><code>CHR</code></dt>
<dd>
<p>a character vector; the chromosome of the
listed SNP.</p>
</dd>
<dt><code>POSITION</code></dt>
<dd>
<p>a numeric vector; the basepair
position of the listed SNP.</p>
</dd>
<dt><code>EFFECT_ALL</code></dt>
<dd>
<p>a character vector; the effect
allele.</p>
</dd>
<dt><code>OTHER_ALL</code></dt>
<dd>
<p>a character vector; the non-effect
allele.</p>
</dd>
<dt><code>N_TOTAL</code></dt>
<dd>
<p>a numeric vector; the sample size.</p>
</dd>
<dt><code>EFF_ALL_FREQ</code></dt>
<dd>
<p>a numeric vector; frequency of
the effect-allele in the dataset.</p>
</dd>
<dt><code>HWE_PVAL</code></dt>
<dd>
<p>a numeric vector; Hardy-Weinberg
(HWE) p-value of the listed alleles.</p>
</dd>
<dt><code>CALLRATE</code></dt>
<dd>
<p>a numeric vector; SNP callrate.</p>
</dd>
<dt><code>EFFECT</code></dt>
<dd>
<p>a numeric vector; effect size of the
listed effect allele.</p>
</dd>
<dt><code>STDERR</code></dt>
<dd>
<p>a numeric vector; standard error of the
effect.</p>
</dd>
<dt><code>PVALUE</code></dt>
<dd>
<p>a numeric vector; p-value of the
effect.</p>
</dd>
<dt><code>IMPUTED</code></dt>
<dd>
<p>a numeric vector; imputation status -
i.e. whether the SNP is genotyped (<code>0</code>) or imputed
(<code>1</code>).</p>
</dd>
<dt><code>IMP_QUALITY</code></dt>
<dd>
<p>a numeric vector; imputation
quality.</p>
</dd>
</dl>
</div>