<div class="container">

<table style="width: 100%;"><tr>
<td>snpprob_collapse</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Collapse genoprob according to pattern</h2>

<h3>Description</h3>

<p>Collapse genoprob according to pattern
</p>


<h3>Usage</h3>

<pre><code class="language-R">snpprob_collapse(
  snpprobs,
  action = c("additive", "add+dom", "non-add", "recessive", "dominant", "basic")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snpprobs</code></td>
<td>
<p>object of class <code>calc_genoprob</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>action</code></td>
<td>
<p>SNP gene action type</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>object of class <code>calc_genoprob</code>
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to snp probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)


dim(snppr[[1]])
dim(snpprob_collapse(snppr, "additive")[[1]])


</code></pre>


</div>