<div class="container">

<table style="width: 100%;"><tr>
<td>subset_scan1</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Subset scan1 output</h2>

<h3>Description</h3>

<p>Subset the output of <code>scan1()</code> by chromosome or column
</p>


<h3>Usage</h3>

<pre><code class="language-R">subset_scan1(x, map = NULL, chr = NULL, lodcolumn = NULL, ...)

## S3 method for class 'scan1'
subset(x, map = NULL, chr = NULL, lodcolumn = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>"scan1"</code> as returned by
<code>scan1()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code>insert_pseudomarkers()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Vector of chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodcolumn</code></td>
<td>
<p>Vector of integers or character strings indicating the LOD
score columns, either as a numeric indexes or column names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Ignored</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Object of class <code>"scan1"</code>, like the input, but subset by chromosome and/or column. See <code>scan1()</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# perform genome scan
out &lt;- scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# pull out chromosome 8
out_c8 &lt;- subset(out, map, chr="8")

# just the second column on chromosome 2
out_c2_spleen &lt;- subset(out, map, "2", "spleen")

# all positions, but just the "liver" column
out_spleen &lt;- subset(out, map, lodcolumn="spleen")
</code></pre>


</div>