<div class="container">

<table style="width: 100%;"><tr>
<td>ggplot_merge_feature</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot of merge_feature object</h2>

<h3>Description</h3>

<p>Merge all SNPs in small region with LOD peaks across multiple phenotype.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ggplot_merge_feature(object, pheno, plot_by = c("pattern", "consequence"), ...)

## S3 method for class 'merge_feature'
autoplot(object, ...)

merge_feature(
  top_snps_tbl,
  snpinfo,
  out_lmm_snps,
  drop = 1.5,
  dropchar = 0,
  exons = gene_exon(top_snps_tbl)
)

## S3 method for class 'merge_feature'
summary(object, sum_type = c("SNP type", "pattern"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>of class <code>merge_feature</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>name of phenotype to be plotted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_by</code></td>
<td>
<p>element to plot by (one of <code>c("pattern","consequence")</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other arguments not used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top_snps_tbl</code></td>
<td>
<p>tbl from <code>top_snps_pattern</code> or <code>top_snps</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpinfo</code></td>
<td>
<p>SNP information table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out_lmm_snps</code></td>
<td>
<p>tbl from <code>scan1</code> on SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop</code></td>
<td>
<p>include LOD scores within <code>drop</code> of max for each phenotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropchar</code></td>
<td>
<p>number of characters to drop on phenames</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exons</code></td>
<td>
<p>table from <code>gene_exon</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sum_type</code></td>
<td>
<p>one of <code>c("SNP type","pattern")</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplot2 object
</p>
<p>tbl with added information on genes and exons
</p>
<p>table summary
</p>


<h3>Author(s)</h3>

<p>Brian S Yandell, <a href="mailto:brian.yandell@wisc.edu">brian.yandell@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">dirpath &lt;- "https://raw.githubusercontent.com/rqtl/qtl2data/master/DOex"

# Read DOex example cross from 'qtl2data'
DOex &lt;- subset(qtl2::read_cross2(file.path(dirpath, "DOex.zip")), chr = "2")


# Download genotype probabilities
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "DOex_genoprobs_2.rds"), tmpfile, quiet=TRUE)
pr &lt;- readRDS(tmpfile)
unlink(tmpfile)

# Download SNP info for DOex from web and read as RDS.
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_snpinfo.rds"), tmpfile, quiet=TRUE)
snpinfo &lt;- readRDS(tmpfile)
unlink(tmpfile)
snpinfo &lt;- dplyr::rename(snpinfo, pos = pos_Mbp)

# Convert to SNP probabilities
snpinfo &lt;- qtl2::index_snps(DOex$pmap, snpinfo)
snppr &lt;- qtl2::genoprob_to_snpprob(pr, snpinfo)

# Scan SNPs.
scan_snppr &lt;- qtl2::scan1(snppr, DOex$pheno)

# Collect top SNPs
top_snps_tbl &lt;- top_snps_pattern(scan_snppr, snpinfo)
summary(top_snps_tbl)

# Download Gene info for DOex from web via RDS
tmpfile &lt;- tempfile()
download.file(file.path(dirpath, "c2_genes.rds"), tmpfile, quiet=TRUE)
gene_tbl &lt;- readRDS(tmpfile)
unlink(tmpfile)

out &lt;- merge_feature(top_snps_tbl, snpinfo, scan_snppr, exons = gene_tbl)
summary(out, "pattern")


</code></pre>


</div>