<div class="container">

<table style="width: 100%;"><tr>
<td>scan1perm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Permutation test for genome scan with a single-QTL model</h2>

<h3>Description</h3>

<p>Permutation test for a enome scan with a single-QTL model by
Haley-Knott regression or a linear mixed model, with possible
allowance for covariates.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scan1perm(
  genoprobs,
  pheno,
  kinship = NULL,
  addcovar = NULL,
  Xcovar = NULL,
  intcovar = NULL,
  weights = NULL,
  reml = TRUE,
  model = c("normal", "binary"),
  n_perm = 1,
  perm_Xsp = FALSE,
  perm_strata = NULL,
  chr_lengths = NULL,
  cores = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genoprobs</code></td>
<td>
<p>Genotype probabilities as calculated by
<code>calc_genoprob()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>A numeric matrix of phenotypes, individuals x phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kinship</code></td>
<td>
<p>Optional kinship matrix, or a list of kinship matrices (one
per chromosome), in order to use the LOCO (leave one chromosome
out) method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>An optional numeric matrix of additive covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xcovar</code></td>
<td>
<p>An optional numeric matrix with additional additive covariates used for
null hypothesis when scanning the X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intcovar</code></td>
<td>
<p>An optional numeric matrix of interactive covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>An optional numeric vector of positive weights for the
individuals. As with the other inputs, it must have <code>names</code>
for individual identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reml</code></td>
<td>
<p>If <code>kinship</code> provided: if <code>reml=TRUE</code>, use
REML; otherwise maximum likelihood.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Indicates whether to use a normal model (least
squares) or binary model (logistic regression) for the phenotype.
If <code>model="binary"</code>, the phenotypes must have values in <code class="reqn">[0, 1]</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_perm</code></td>
<td>
<p>Number of permutation replicates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm_Xsp</code></td>
<td>
<p>If TRUE, do separate permutations for the autosomes
and the X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm_strata</code></td>
<td>
<p>Vector of strata, for a stratified permutation
test. Should be named in the same way as the rows of
<code>pheno</code>. The unique values define the strata.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_lengths</code></td>
<td>
<p>Lengths of the chromosomes; needed only if
<code>perm_Xsp=TRUE</code>. See <code>chr_lengths()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code>parallel::detectCores()</code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code>parallel::makeCluster()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional control parameters; see Details.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>kinship</code> is not provided, so that analysis proceeds by
Haley-Knott regression, we permute the rows of the phenotype data;
the same permutations are also applied to the rows of the
covariates (<code>addcovar</code>, <code>Xcovar</code>, and <code>intcovar</code>)
are permuted.
</p>
<p>If <code>kinship</code> is provided, we instead permute the rows of the
genotype data and fit an LMM with the same residual heritability
(estimated under the null hypothesis of no QTL).
</p>
<p>If <code>Xcovar</code> is provided and <code>perm_strata=NULL</code>, we do a
stratified permutation test with the strata defined by the rows of
<code>Xcovar</code>. If a simple permutation test is desired, provide
<code>perm_strata</code> that is a vector containing a single repeated
value.
</p>
<p>The <code>...</code> argument can contain several additional control
parameters; suspended for simplicity (or confusion, depending on
your point of view). <code>tol</code> is used as a tolerance value for linear
regression by QR decomposition (in determining whether columns are
linearly dependent on others and should be omitted); default
<code>1e-12</code>. <code>maxit</code> is the maximum number of iterations for
converence of the iterative algorithm used when <code>model=binary</code>.
<code>bintol</code> is used as a tolerance for converence for the iterative
algorithm used when <code>model=binary</code>. <code>eta_max</code> is the maximum value
for the "linear predictor" in the case <code>model="binary"</code> (a bit of a
technicality to avoid fitted values exactly at 0 or 1).
</p>


<h3>Value</h3>

<p>If <code>perm_Xsp=FALSE</code>, the result is matrix of
genome-wide maximum LOD scores, permutation replicates x
phenotypes. If <code>perm_Xsp=TRUE</code>, the result is a list of
two matrices, one for the autosomes and one for the X
chromosome. The object is given class <code>"scan1perm"</code>.
</p>


<h3>References</h3>

<p>Churchill GA, Doerge RW (1994) Empirical threshold
values for quantitative trait mapping. Genetics 138:963–971.
</p>
<p>Manichaikul A, Palmer AA, Sen S, Broman KW (2007) Significance
thresholds for quantitative trait locus mapping under selective
genotyping. Genetics 177:1963–1966.
</p>
<p>Haley CS, Knott SA (1992) A simple regression method for mapping
quantitative trait loci in line crosses using flanking markers.
Heredity 69:315–324.
</p>
<p>Kang HM, Zaitlen NA, Wade CM, Kirby A, Heckerman D, Daly MJ, Eskin
E (2008) Efficient control of population structure in model
organism association mapping. Genetics 178:1709–1723.
</p>


<h3>See Also</h3>

<p><code>scan1()</code>, <code>chr_lengths()</code>, <code>mat2strata()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))


# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# strata for permutations
perm_strata &lt;- mat2strata(Xcovar)

# permutations with genome scan (just 3 replicates, for illustration)
operm &lt;- scan1perm(probs, pheno, addcovar=covar, Xcovar=Xcovar,
                   n_perm=3, perm_strata=perm_strata)
summary(operm)

# leave-one-chromosome-out kinship matrices
kinship &lt;- calc_kinship(probs, "loco")

# permutations of genome scan with a linear mixed model

operm_lmm &lt;- scan1perm(probs, pheno, kinship, covar, Xcovar, n_perm=3,
                       perm_Xsp=TRUE, perm_strata=perm_strata,
                       chr_lengths=chr_lengths(map))
summary(operm_lmm)


</code></pre>


</div>