<div class="container">

<table style="width: 100%;"><tr>
<td>recode_snps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Recode SNPs by major allele</h2>

<h3>Description</h3>

<p>For multi-parent populations with founder genotypes, recode the raw
SNP genotypes so that <code>1</code> means homozygous for the major allele in the
founders.
</p>


<h3>Usage</h3>

<pre><code class="language-R">recode_snps(cross)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>Object of class <code>"cross2"</code>. For details, see the
<a href="https://kbroman.org/qtl2/assets/vignettes/developer_guide.html">R/qtl2 developer guide</a>.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>The input cross object with the raw SNP genotypes recoded so that
<code>1</code> is homozygous for the major alleles in the founders.
</p>


<h3>See Also</h3>

<p><code>calc_raw_founder_maf()</code>, <code>calc_raw_maf()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# load example data and calculate genotype probabilities
file &lt;- paste0("https://raw.githubusercontent.com/rqtl/",
               "qtl2data/main/DOex/DOex.zip")
DOex &lt;- read_cross2(file)
DOex &lt;- recode_snps(DOex)

## End(Not run)
</code></pre>


</div>