<div class="container">

<table style="width: 100%;"><tr>
<td>ggplot_scan1</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot a genome scan</h2>

<h3>Description</h3>

<p>Plot LOD curves for a genome scan
</p>
<p>Plot LOD curves for a genome scan
</p>


<h3>Usage</h3>

<pre><code class="language-R">ggplot_scan1(
  object,
  map,
  lodcolumn = 1,
  chr = NULL,
  gap = 25,
  bgcolor = "gray90",
  altbgcolor = "gray85",
  ...
)

## S3 method for class 'scan1'
autoplot(object, ...)

ggplot_scan1_internal(
  map,
  lod,
  gap = 25,
  col = NULL,
  shape = NULL,
  pattern = NULL,
  facet = NULL,
  patterns = c("none", "all", "hilit"),
  chrName = "Chr",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>Output of <code>scan1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Map of pseudomarker locations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodcolumn</code></td>
<td>
<p>LOD score column to plot (a numeric index, or a
character string for a column name). One or more value(s) allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Selected chromosomes to plot; a vector of character
strings.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>
<p>Gap between chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bgcolor</code></td>
<td>
<p>Background color for the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>altbgcolor</code></td>
<td>
<p>Background color for alternate chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional graphics parameters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lod</code></td>
<td>
<p>Matrix of lod (or other) values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>Colors for points or lines, with labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shape</code></td>
<td>
<p>Shapes for points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pattern</code></td>
<td>
<p>Use to group values for plotting (default = <code>NULL</code>); typically provided by <code>plot_snpasso</code> internal routine.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>facet</code></td>
<td>
<p>Plot facets if multiple phenotypes and group provided (default = <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>patterns</code></td>
<td>
<p>Connect SDP patterns: one of <code>c("none","all","hilit")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrName</code></td>
<td>
<p>Add prefix chromosome name (default <code>"Chr"</code>).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>None.
</p>


<h3>See Also</h3>

<p><code>ggplot_coef</code>, <code>ggplot_snpasso</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load qtl2 package for data and genoprob calculation
library(qtl2)

# read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# perform genome scan
out &lt;- scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# plot the results for selected chromosomes
chr &lt;- c(2,7,8,9,15,16)
ggplot_scan1(out, map, lodcolumn=1:2, chr=chr, col=c("darkslateblue","violetred"),
     legend.position=c(0.1,0.9))

# plot just one chromosome
ggplot_scan1(out, map, chr=8, lodcolumn=1:2, col=c("darkblue","violetred"))

# can also use autoplot from ggplot2
# lodcolumn can also be a column name
library(ggplot2)
autoplot(out, map, chr=8, lodcolumn=c("liver","spleen"), col=c("darkblue","violetred"))
</code></pre>


</div>