<div class="container">

<table style="width: 100%;"><tr>
<td>scan_multi_onechr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform multivariate, one-QTL model fitting for markers on one chromosome</h2>

<h3>Description</h3>

<p>'scan_multi_onechr' calculates log likelihood for d-variate phenotype model fits. Inputted parameter 'start_snp' indicates where in the 'probs' object to start the scan.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scan_multi_onechr(
  probs,
  pheno,
  kinship = NULL,
  addcovar = NULL,
  start_snp = 1,
  n_snp = dim(probs)[3],
  max_iter = 10000,
  max_prec = 1/1e+08,
  cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>probs</code></td>
<td>
<p>an array of founder allele probabilities for a single chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>a matrix of phenotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kinship</code></td>
<td>
<p>a kinship matrix for one chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>a matrix, n subjects by c additive covariates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start_snp</code></td>
<td>
<p>index of where to start the scan within probs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_snp</code></td>
<td>
<p>the number of (consecutive) markers to include in the scan</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_iter</code></td>
<td>
<p>maximum number of iterations for EM algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_prec</code></td>
<td>
<p>stepwise precision for EM algorithm. EM stops once incremental difference in log likelihood is less than max_prec</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>number of cores for parallelization</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a tibble with d + 1 columns. First d columns indicate the genetic data (by listing the marker ids) used in the design matrix; last is log10 likelihood
</p>


<h3>References</h3>

<p>Knott SA, Haley CS (2000) Multitrait
least squares for quantitative trait loci detection.
Genetics 156: 899â€“911.
</p>
<p>Jiang C, Zeng ZB (1995) Multiple trait analysis
of genetic mapping for quantitative trait loci.
Genetics 140: 1111-1127.
</p>
<p>Zhou X, Stephens M (2014) Efficient multivariate linear
mixed model algorithms for genome-wide association studies.
Nature methods 11:407-409.
</p>
<p>Broman KW, Gatti DM, Simecek P, Furlotte NA, Prins P, Sen S, Yandell BS, Churchill GA (2019)
R/qtl2: software for mapping quantitative trait loci with high-dimensional data and
multi-parent populations. GENETICS https://www.genetics.org/content/211/2/495.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read data
n &lt;- 50
pheno &lt;- matrix(rnorm(2 * n), ncol = 2)
rownames(pheno) &lt;- paste0("s", 1:n)
colnames(pheno) &lt;- paste0("tr", 1:2)
probs &lt;- array(dim = c(n, 2, 5))
probs[ , 1, ] &lt;- rbinom(n * 5, size = 1, prob = 0.2)
probs[ , 2, ] &lt;- 1 - probs[ , 1, ]
rownames(probs) &lt;- paste0("s", 1:n)
colnames(probs) &lt;- LETTERS[1:2]
dimnames(probs)[[3]] &lt;- paste0("m", 1:5)
scan_multi_onechr(probs = probs, pheno = pheno, kinship = NULL, cores = 1)

</code></pre>


</div>