<div class="container">

<table style="width: 100%;"><tr>
<td>find_ibd_segments</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find IBD segments for a set of strains</h2>

<h3>Description</h3>

<p>Find IBD segments (regions with a lot of shared SNP genotypes) for
a set of strains
</p>


<h3>Usage</h3>

<pre><code class="language-R">find_ibd_segments(geno, map, min_lod = 15, error_prob = 0.001, cores = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>List of matrices of founder genotypes. The matrices
correspond to the genotypes on chromosomes and are arrayed as
founders x markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>List of vectors of marker positions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_lod</code></td>
<td>
<p>Threshold for minimum LOD score for a segment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error_prob</code></td>
<td>
<p>Genotyping error/mutation probability</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code>parallel::detectCores()</code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code>parallel::makeCluster()</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For each strain pair on each chromosome, we consider all
marker intervals and calculate a LOD score comparing the two
hypotheses: that the strains are IBD in the interval, vs. that
they are not. We assume that the two strains are homozygous at
all markers, and use the model from Broman and Weber (1999),
which assumes linkage equilibrium between markers and uses a
simple model for genotype frequencies in the presence of genotyping
errors or mutations.
</p>
<p>Note that inference of IBD segments is heavily dependent on how
SNPs were chosen to be genotyped. (For example, were the SNPs ascertained
based on their polymorphism between a particular strain pair?)
</p>


<h3>Value</h3>

<p>A data frame whose rows are IBD segments and whose columns
are:
</p>

<ul>
<li>
<p> Strain 1
</p>
</li>
<li>
<p> Strain 2
</p>
</li>
<li>
<p> Chromosome
</p>
</li>
<li>
<p> Left marker
</p>
</li>
<li>
<p> Right marker
</p>
</li>
<li>
<p> Left position
</p>
</li>
<li>
<p> Right position
</p>
</li>
<li>
<p> Left marker index
</p>
</li>
<li>
<p> Right marker index
</p>
</li>
<li>
<p> Interval length
</p>
</li>
<li>
<p> Number of markers
</p>
</li>
<li>
<p> Number of mismatches
</p>
</li>
<li>
<p> LOD score
</p>
</li>
</ul>
<h3>References</h3>

<p>Broman KW, Weber JL (1999) Long homozygous chromosomal segments in
reference families from the Centre d’Étude du Polymorphisme Humain.
Am J Hum Genet 65:1493–1500.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# load DO data from Recla et al. (2014) Mamm Genome 25:211-222.
recla &lt;- read_cross2("https://raw.githubusercontent.com/rqtl/qtl2data/main/DO_Recla/recla.zip")

# grab founder genotypes and physical map
fg &lt;- recla$founder_geno
pmap &lt;- recla$pmap

# find shared segments
(segs &lt;- find_ibd_segments(fg, pmap, min_lod=10, error_prob=0.0001))

## End(Not run)

</code></pre>


</div>