<div class="container">

<table style="width: 100%;"><tr>
<td>calc_genoprob</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate conditional genotype probabilities</h2>

<h3>Description</h3>

<p>Uses a hidden Markov model to calculate the probabilities of the
true underlying genotypes given the observed multipoint marker
data, with possible allowance for genotyping errors.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calc_genoprob(
  cross,
  map = NULL,
  error_prob = 0.0001,
  map_function = c("haldane", "kosambi", "c-f", "morgan"),
  lowmem = FALSE,
  quiet = TRUE,
  cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>Object of class <code>"cross2"</code>. For details, see the
<a href="https://kbroman.org/qtl2/assets/vignettes/developer_guide.html">R/qtl2 developer guide</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Genetic map of markers. May include pseudomarker
locations (that is, locations that are not within the marker
genotype data). If NULL, the genetic map in <code>cross</code> is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error_prob</code></td>
<td>
<p>Assumed genotyping error probability</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map_function</code></td>
<td>
<p>Character string indicating the map function to
use to convert genetic distances to recombination fractions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lowmem</code></td>
<td>
<p>If <code>FALSE</code>, split individuals into groups with
common sex and crossinfo and then precalculate the transition
matrices for a chromosome; potentially a lot faster but using more
memory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>If <code>FALSE</code>, print progress messages.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code>parallel::detectCores()</code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code>parallel::makeCluster()</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Let <code class="reqn">O_k</code> denote the observed marker genotype at position
<code class="reqn">k</code>, and <code class="reqn">g_k</code> denote the corresponding true underlying
genotype.
</p>
<p>We use the forward-backward equations to calculate
<code class="reqn">\alpha_{kv} = \log Pr(O_1, \ldots, O_k, g_k = v)</code>
and
<code class="reqn">\beta_{kv} = \log Pr(O_{k+1}, \ldots, O_n | g_k = v)</code>
</p>
<p>We then obtain
<code class="reqn">Pr(g_k | O_1, \ldots, O_n) = \exp(\alpha_{kv} + \beta_{kv}) / s</code>
where
<code class="reqn">s = \sum_v \exp(\alpha_{kv} + \beta_{kv})</code>
</p>


<h3>Value</h3>

<p>An object of class <code>"calc_genoprob"</code>: a list of three-dimensional arrays of probabilities,
individuals x genotypes x positions. (Note that the arrangement is
different from R/qtl.) Also contains four attributes:
</p>

<ul>
<li> <p><code>crosstype</code> - The cross type of the input <code>cross</code>.
</p>
</li>
<li> <p><code>is_x_chr</code> - Logical vector indicating whether chromosomes
are to be treated as the X chromosome or not, from input <code>cross</code>.
</p>
</li>
<li> <p><code>alleles</code> - Vector of allele codes, from input
<code>cross</code>.
</p>
</li>
<li> <p><code>alleleprobs</code> - Logical value (<code>FALSE</code>) that
indicates whether the probabilities are compressed to allele
probabilities, as from <code>genoprob_to_alleleprob()</code>.
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>insert_pseudomarkers()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">grav2 &lt;- read_cross2(system.file("extdata", "grav2.zip", package="qtl2"))
gmap_w_pmar &lt;- insert_pseudomarkers(grav2$gmap, step=1)
probs &lt;- calc_genoprob(grav2, gmap_w_pmar, error_prob=0.002)
</code></pre>


</div>