<div class="container">

<table style="width: 100%;"><tr>
<td>snpinfo_to_map</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert SNP info to map</h2>

<h3>Description</h3>

<p>Convert SNP info to map
</p>


<h3>Usage</h3>

<pre><code class="language-R">snpinfo_to_map(snpinfo)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>snpinfo</code></td>
<td>
<p>Data frame with SNP information with the following
columns (the last three are generally derived from with
<code>index_snps</code>):
</p>

<ul>
<li> <p><code>chr</code> - Character string or factor with chromosome
</p>
</li>
<li> <p><code>pos</code> - Position (in same units as in the <code>"map"</code>
attribute in <code>genoprobs</code>.
</p>
</li>
<li> <p><code>sdp</code> - Strain distribution pattern: an integer, between
1 and <code class="reqn">2^n - 2</code> where <code class="reqn">n</code> is the number of strains, whose
binary encoding indicates the founder genotypes
</p>
</li>
<li> <p><code>snp</code> - Character string with SNP identifier (if
missing, the rownames are used).
</p>
</li>
<li> <p><code>index</code> - Indices that indicate equivalent
groups of SNPs.
</p>
</li>
<li> <p><code>intervals</code> - Indexes that indicate which marker
intervals the SNPs reside.
</p>
</li>
<li> <p><code>on_map</code> - Indicate whether SNP coincides with a marker
in the <code>genoprobs</code>
</p>
</li>
</ul>
</td>
</tr></table>
<h3>Value</h3>

<p>map as list of vectors of marker positions.
</p>


</div>