<div class="container">

<table style="width: 100%;"><tr>
<td>maxlod</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Overall maximum LOD score</h2>

<h3>Description</h3>

<p>Find overall maximum LOD score in genome scan results, across all positions and columns.
</p>


<h3>Usage</h3>

<pre><code class="language-R">maxlod(scan1_output, map = NULL, chr = NULL, lodcolumn = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scan1_output</code></td>
<td>
<p>An object of class <code>"scan1"</code> as returned by
<code>scan1()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code>insert_pseudomarkers()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector of chromosomes to consider.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodcolumn</code></td>
<td>
<p>An integer or character string indicating the LOD
score column, either as a numeric index or column name.
If <code>NULL</code>, return maximum for all columns.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A single number: the maximum LOD score across all columns and positions for
the selected chromosomes.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# perform genome scan
out &lt;- scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# overall maximum
maxlod(out)

# maximum on chromosome 2
maxlod(out, map, "2")
</code></pre>


</div>