<div class="container">

<table style="width: 100%;"><tr>
<td>bayes_int</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Bayes credible intervals</h2>

<h3>Description</h3>

<p>Calculate Bayes credible intervals for a single LOD curve on a single
chromosome, with the ability to identify intervals for multiple LOD
peaks.
</p>


<h3>Usage</h3>

<pre><code class="language-R">bayes_int(
  scan1_output,
  map,
  chr = NULL,
  lodcolumn = 1,
  threshold = 0,
  peakdrop = Inf,
  prob = 0.95,
  expand2markers = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scan1_output</code></td>
<td>
<p>An object of class <code>"scan1"</code> as returned by
<code>scan1()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code>insert_pseudomarkers()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Chromosome ID to consider (must be a single value).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodcolumn</code></td>
<td>
<p>LOD score column to consider (must be a single value).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>Minimum LOD score for a peak.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peakdrop</code></td>
<td>
<p>Amount that the LOD score must drop between peaks,
if multiple peaks are to be defined on a chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prob</code></td>
<td>
<p>Nominal coverage for the interval.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expand2markers</code></td>
<td>
<p>If TRUE, QTL intervals are expanded so
that their endpoints are at genetic markers.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We identify a set of peaks defined as local maxima that
exceed the specified <code>threshold</code>, with the requirement that
the LOD score must have dropped by at least <code>peakdrop</code> below
the lowest of any two adjacent peaks.
</p>
<p>At a given peak, if there are ties, with multiple positions jointly
achieving the maximum LOD score, we take the average of these
positions as the location of the peak.
</p>
<p>The default is to use <code>threshold=0</code>, <code>peakdrop=Inf</code>, and
<code>prob=0.95</code>. We then return results a single peak, no matter the
maximum LOD score, and give a 95% Bayes credible interval.
</p>


<h3>Value</h3>

<p>A matrix with three columns:
</p>

<ul>
<li> <p><code>ci_lo</code> - lower bound of interval
</p>
</li>
<li> <p><code>pos</code> - peak position
</p>
</li>
<li> <p><code>ci_hi</code> - upper bound of interval
</p>
</li>
</ul>
<p>Each row corresponds to a different peak.
</p>


<h3>See Also</h3>

<p><code>lod_int()</code>, <code>find_peaks()</code>, <code>scan1()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))


# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# perform genome scan
out &lt;- scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# 95% Bayes credible interval for QTL on chr 7, first phenotype
bayes_int(out, map, chr=7, lodcolum=1)
</code></pre>


</div>