<div class="container">

<table style="width: 100%;"><tr>
<td>plot_genoprobcomp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot comparison of two sets of genotype probabilities</h2>

<h3>Description</h3>

<p>Plot a comparison of two sets of genotype probabilities for one individual on one chromosome, as a heat map.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_genoprobcomp(
  probs1,
  probs2,
  map,
  ind = 1,
  chr = NULL,
  geno = NULL,
  threshold = 0,
  n_colors = 256,
  swap_axes = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>probs1</code></td>
<td>
<p>Genotype probabilities (as produced by <code>calc_genoprob()</code>)
or allele dosages (as produced by <code>genoprob_to_alleleprob()</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probs2</code></td>
<td>
<p>A second set of genotype probabilities, just like <code>probs1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Marker map (a list of vectors of marker positions).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind</code></td>
<td>
<p>Individual to plot, either a numeric index or an ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Selected chromosome to plot; a single character string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>Optional vector of genotypes or alleles to be shown
(vector of integers or character strings)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>Threshold for genotype probabilities; only genotypes that achieve
this value somewhere on the chromosome (in one or the other set of probabilities) will be shown.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_colors</code></td>
<td>
<p>Number of colors in each color scale.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>swap_axes</code></td>
<td>
<p>If TRUE, swap the axes, so that the genotypes are
on the x-axis and the chromosome position is on the y-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional graphics parameters passed to <code>graphics::image()</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We plot the first set of probabilities in the range white to blue
and the second set in the range white to red and attempt to combine
them, for colors that are white, some amount of blue or red, or
where both are large something like blackish purple.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>See Also</h3>

<p><code>plot_genoprob()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))
iron &lt;- iron[228,"1"]   # subset to one individual on chr 1
map &lt;- insert_pseudomarkers(iron$gmap, step=5)

# introduce genotype error and look at difference in genotype probabilities
pr_ne &lt;- calc_genoprob(iron, map, error_prob=0.002)
iron$geno[[1]][1,2] &lt;- 3
pr_e &lt;- calc_genoprob(iron, map, error_prob=0.002)

# image of probabilities + comparison

par(mfrow=c(3,1))
plot_genoprob(pr_ne, map, main="No error")
plot_genoprob(pr_e, map, main="With an error")
plot_genoprobcomp(pr_ne, pr_e, map, main="Comparison")


</code></pre>


</div>