<div class="container">

<table style="width: 100%;"><tr>
<td>guess_phase</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Guess phase of imputed genotypes</h2>

<h3>Description</h3>

<p>Turn imputed genotypes into phased genotypes along chromosomes by
attempting to pick the phase that leads to the fewest recombination
events.
</p>


<h3>Usage</h3>

<pre><code class="language-R">guess_phase(cross, geno, deterministic = FALSE, cores = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>Object of class <code>"cross2"</code>. For details, see the
<a href="https://kbroman.org/qtl2/assets/vignettes/developer_guide.html">R/qtl2 developer guide</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>Imputed genotypes, as a list of matrices, as from <code>maxmarg()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deterministic</code></td>
<td>
<p>If TRUE, preferentially put smaller allele first when there's uncertainty.
If FALSE, the order of alleles is random in such cases.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code>parallel::detectCores()</code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code>parallel::makeCluster()</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We randomly assign the pair of alleles at the first locus
to two haplotypes, and then work left to right, assigning
alleles to haplotypes one locus at a time seeking the fewest
recombination events. The results are subject to arbitrary and
random choices. For example, to the right of a homozygous
region, either orientation is equally reasonable.
</p>


<h3>Value</h3>

<p>If input cross is phase-known (e.g., recombinant inbred lines),
the output will be the input <code>geno</code>. Otherwise, the output
will be a list of three-dimensional arrays of imputed
genotypes, individual x position x haplotype (1/2).
</p>


<h3>See Also</h3>

<p><code>maxmarg()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))
gmap &lt;- insert_pseudomarkers(iron$gmap, step=1)
probs &lt;- calc_genoprob(iron, gmap, error_prob=0.002)
imp_geno &lt;- maxmarg(probs)
ph_geno &lt;- guess_phase(iron, imp_geno)
</code></pre>


</div>