<div class="container">

<table style="width: 100%;"><tr>
<td>plot_genoprob</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot genotype probabilities for one individual on one chromosome.</h2>

<h3>Description</h3>

<p>Plot the genotype probabilities for one individual on one chromosome, as a heat map.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_genoprob(
  probs,
  map,
  ind = 1,
  chr = NULL,
  geno = NULL,
  color_scheme = c("gray", "viridis"),
  col = NULL,
  threshold = 0,
  swap_axes = FALSE,
  ...
)

## S3 method for class 'calc_genoprob'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>probs</code></td>
<td>
<p>Genotype probabilities (as produced by <code>calc_genoprob()</code>)
or allele dosages (as produced by <code>genoprob_to_alleleprob()</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>Marker map (a list of vectors of marker positions).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind</code></td>
<td>
<p>Individual to plot, either a numeric index or an ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Selected chromosome to plot; a single character string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>Optional vector of genotypes or alleles to be shown
(vector of integers or character strings)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_scheme</code></td>
<td>
<p>Color scheme for the heatmap (ignored if <code>col</code> is provided).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>Optional vector of colors for the heatmap.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>Threshold for genotype probabilities; only genotypes that achieve
this value somewhere on the chromosome will be shown.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>swap_axes</code></td>
<td>
<p>If TRUE, swap the axes, so that the genotypes are
on the x-axis and the chromosome position is on the y-axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional graphics parameters passed to <code>graphics::image()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Genotype probabilities (as produced by
<code>calc_genoprob()</code>) or allele dosages (as produced by
<code>genoprob_to_alleleprob()</code>). (For the S3 type plot
function, this has to be called <code>x</code>.)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>None.
</p>


<h3>Hidden graphics parameters</h3>

<p>A number of graphics parameters can be passed via <code>...</code>. For
example, <code>hlines</code>, <code>hlines_col</code>, <code>hlines_lwd</code>, and <code>hlines_lty</code> to
control the horizontal grid lines. (Use <code>hlines=NA</code> to avoid
plotting horizontal grid lines.) Similarly <code>vlines</code>, <code>vlines_col</code>,
<code>vlines_lwd</code>, and <code>vlines_lty</code> for vertical grid lines. You can
also use many standard graphics parameters like <code>xlab</code> and <code>xlim</code>.
These are not included as formal parameters in order to avoid
cluttering the function definition.
</p>


<h3>See Also</h3>

<p><code>plot_genoprobcomp()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load data and calculate genotype probabilities
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))
iron &lt;- iron[,"2"] # subset to chr 2
map &lt;- insert_pseudomarkers(iron$gmap, step=1)
pr &lt;- calc_genoprob(iron, map, error_prob=0.002)

# plot the probabilities for the individual labeled "262"
#  (white = 0, black = 1)
plot_genoprob(pr, map, ind="262")

# change the x-axis label
plot_genoprob(pr, map, ind="262", xlab="Position (cM)")

# swap the axes so that the chromosome runs vertically
plot_genoprob(pr, map, ind="262", swap_axes=TRUE, ylab="Position (cM)")

# This is more interesting for a Diversity Outbred mouse example
## Not run: 
file &lt;- paste0("https://raw.githubusercontent.com/rqtl/",
               "qtl2data/main/DOex/DOex.zip")
DOex &lt;- read_cross2(file)
# subset to chr 2 and X and individuals labeled "232" and "256"
DOex &lt;- DOex[c("232", "256"), c("2", "X")]
pr &lt;- calc_genoprob(DOex, error_prob=0.002)
# plot individual "256" on chr 2 (default is to pick first chr in the probs)
plot_genoprob(pr, DOex$pmap, ind="256")

# omit states that never have probability &gt;= 0.5
plot_genoprob(pr, DOex$pmap, ind="256", threshold=0.05)

# X chr male 232: just show the AY-HY genotype probabilities
plot_genoprob(pr, DOex$pmap, ind="232", chr="X", geno=paste0(LETTERS[1:8], "Y"))
# could also indicate genotypes by number
plot_genoprob(pr, DOex$pmap, ind="232", chr="X", geno=37:44)
# and can use negative indexes
plot_genoprob(pr, DOex$pmap, ind="232", chr="X", geno=-(1:36))

# X chr female 256: just show the first 36 genotype probabilities
plot_genoprob(pr, DOex$pmap, ind="256", chr="X", geno=1:36)

# again, can give threshold to omit genotypes whose probabilities never reach that threshold
plot_genoprob(pr, DOex$pmap, ind="256", chr="X", geno=1:36, threshold=0.5)

# can also look at the allele dosages
apr &lt;- genoprob_to_alleleprob(pr)
plot_genoprob(apr, DOex$pmap, ind="232")

## End(Not run)

</code></pre>


</div>