<div class="container">

<table style="width: 100%;"><tr>
<td>max_scan1</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find position with maximum LOD score</h2>

<h3>Description</h3>

<p>Return data frame with the positions having maximum LOD score for a
particular LOD score column
</p>


<h3>Usage</h3>

<pre><code class="language-R">max_scan1(
  scan1_output,
  map = NULL,
  lodcolumn = 1,
  chr = NULL,
  na.rm = TRUE,
  ...
)

## S3 method for class 'scan1'
max(scan1_output, map = NULL, lodcolumn = 1, chr = NULL, na.rm = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scan1_output</code></td>
<td>
<p>An object of class <code>"scan1"</code> as returned by
<code>scan1()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>A list of vectors of marker positions, as produced by
<code>insert_pseudomarkers()</code>. Can also be an indexed SNP info table,
as from <code>index_snps()</code> or <code>scan1snps()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodcolumn</code></td>
<td>
<p>An integer or character string indicating the LOD
score column, either as a numeric index or column name.
If <code>NULL</code>, return maximum for all columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector of chromosomes to consider.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.rm</code></td>
<td>
<p>Ignored (take to be TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Ignored</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>map</code> is NULL, the genome-wide maximum LOD score for the selected column is returned.
If also <code>lodcolumn</code> is NULL, you get a vector with the maximum LOD for each column.
</p>
<p>If <code>map</code> is provided, the return value is a data.frame with three columns: chr, pos, and lod score.
But if <code>lodcolumn</code> is NULL, you get the maximum for each lod score column, in the format provided by
<code>find_peaks()</code>, so a data.frame with five columns: lodindex, lodcolumn, chr, pos, and lod.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># read data
iron &lt;- read_cross2(system.file("extdata", "iron.zip", package="qtl2"))

# insert pseudomarkers into map
map &lt;- insert_pseudomarkers(iron$gmap, step=1)

# calculate genotype probabilities
probs &lt;- calc_genoprob(iron, map, error_prob=0.002)

# grab phenotypes and covariates; ensure that covariates have names attribute
pheno &lt;- iron$pheno
covar &lt;- match(iron$covar$sex, c("f", "m")) # make numeric
names(covar) &lt;- rownames(iron$covar)
Xcovar &lt;- get_x_covar(iron)

# perform genome scan
out &lt;- scan1(probs, pheno, addcovar=covar, Xcovar=Xcovar)

# maximum of first column
max(out, map)

# maximum of spleen column
max(out, map, lodcolumn="spleen")

# maximum of first column on chr 2
max(out, map, chr="2")
</code></pre>


</div>