<div class="container">

<table style="width: 100%;"><tr>
<td>inferFounderHap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Crude reconstruction of founder haplotypes in multi-parent RIL</h2>

<h3>Description</h3>

<p>Uses groups of adjacent markers to infer the founder haplotypes in SNP
data on multi-parent recombinant inbred lines.
</p>


<h3>Usage</h3>

<pre><code class="language-R">inferFounderHap(cross, chr, max.n.markers=15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>cross</code>. See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Indicator of chromosome to consider.  If multiple
chromosomes are selected, only the first is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.n.markers</code></td>
<td>
<p>Maximum number of adjacent markers to consider.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We omit SNPs for which any of the founders are missing.
</p>
<p>We then consider groups of adjacent SNPs, looking for founder
haplotypes that are unique; RIL sharing such a unique haplotype are
then inferred to have that founder's DNA.
</p>
<p>We consider each marker as the center of a haplotype, and consider
haplotypes of size 1, 3, 5, ..., <code>max.n.markers</code>.  We end the
extension of the haplotypes when all founders have a unique haplotype.
</p>


<h3>Value</h3>

<p>A matrix of dimension <code>nind(cross)</code> <code class="reqn">\times</code> no. markers,
with the inferred founder origin for each line at each marker.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>See Also</h3>

 <p><code>sim.geno</code>, <code>calc.genoprob</code>,
<code>fill.geno</code>, <code>argmax.geno</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">map &lt;- sim.map(100, n.mar=101, include.x=FALSE, eq.spacing=TRUE)
founderGeno &lt;- simFounderSnps(map, "8")
ril &lt;- sim.cross(map, n.ind=10, type="ri8sib", founderGeno=founderGeno)

h &lt;- inferFounderHap(ril, max.n.markers=11)
mean(!is.na(h)) # proportion inferred
plot(map[[1]], h[1,], ylim=c(0.5, 8.5), xlab="Position", ylab="Genotype")
</code></pre>


</div>