<div class="container">

<table style="width: 100%;"><tr>
<td>inferredpartitions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify inferred partitions in mapping QTL to a phylogenetic tree</h2>

<h3>Description</h3>

<p>Identify the inferred partitions for a chromosome from the results of scanPhyloQTL.
</p>


<h3>Usage</h3>

<pre><code class="language-R">inferredpartitions(output, chr, lodthreshold, probthreshold=0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>An object output by the function
<code>scanPhyloQTL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>A character string indicating the chromosome to
consider. (It can also be a number, but it's then converted to a
character string.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodthreshold</code></td>
<td>
<p>LOD threshold; if maximum LOD score is less than
this, the null model is considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probthreshold</code></td>
<td>
<p>Threshold on posterior probabilities.  See Details
below.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We consider a single chromosome, and take the maximum LOD score for
each partition on that chromosome.  The presence of a QTL is inferred
if at least one partition has LOD score greater than
<code>lodthreshold</code>.  In this case, we then convert the LOD scores for
the partitions to approximate posterior probabilities by taking
<code class="reqn">10^{LOD}</code> and then rescaling them to sum to 1.
These are sorted from largest to smallest, and we
then take as the inferred partitions the smallest set whose posterior
probabilities cumulatively add up to at least <code>probthreshold</code>.
</p>


<h3>Value</h3>

<p>A vector of character strings.  If the null model (no QTL) is
inferred, the output is <code>"null"</code>.  Otherwise, it is the set of
inferred partitions.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a></p>


<h3>References</h3>

<p>Broman, K. W., Kim, S., An\'e, C. and Payseur, B. A.  Mapping
quantitative trait loci to a phylogenetic tree.  In preparation.
</p>


<h3>See Also</h3>

 <p><code>scanPhyloQTL</code>, <code>plot.scanPhyloQTL</code>,
<code>summary.scanPhyloQTL</code>, <code>max.scanPhyloQTL</code>,
<code>simPhyloQTL</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
# example map; drop X chromosome
data(map10)
map10 &lt;- map10[1:19]


# simulate data
x &lt;- simPhyloQTL(4, partition="AB|CD", crosses=c("AB", "AC", "AD"),
                 map=map10, n.ind=150,
                 model=c(1, 50, 0.5, 0))

# run calc.genoprob on each cross
## Not run: x &lt;- lapply(x, calc.genoprob, step=2)


# scan genome, at each position trying all possible partitions
out &lt;- scanPhyloQTL(x, method="hk")

# inferred partitions
inferredpartitions(out, chr=3, lodthreshold=3)

# inferred partitions with prob'y threshold = 0.95
inferredpartitions(out, chr=3, lodthreshold=3, probthreshold=0.95)
</code></pre>


</div>