<div class="container">

<table style="width: 100%;"><tr>
<td>simPhyloQTL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate a set of intercrosses for a single diallelic QTL</h2>

<h3>Description</h3>

<p>Simulate a set of intercrosses with a single diallelic QTL.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simPhyloQTL(n.taxa=3, partition, crosses, map, n.ind=100, model,
            error.prob=0, missing.prob=0, partial.missing.prob=0,
            keep.qtlgeno=FALSE, keep.errorind=TRUE, m=0, p=0,
        map.function=c("haldane","kosambi","c-f","morgan"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.taxa</code></td>
<td>
<p>Number of taxa (i.e., strains).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partition</code></td>
<td>
<p>A vector of character strings of the form "AB|CD" or "A|BCD"
indicating, for each QTL, which taxa have which allele.  If missing,
simulate under the null hypothesis of no QTL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crosses</code></td>
<td>
<p>A vector of character strings indicating the crosses to
do (for the form "AB", "AC", etc.).  These will be sorted and then only
unique ones used.  If missing, all crosses will be simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>A list whose components are vectors containing the marker
locations on each of the chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.ind</code></td>
<td>
<p>The number of individuals in each cross.  If length 1, all
crosses will have the same number of individuals; otherwise the length
should be the same as <code>crosses</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>A matrix where each row corresponds to a
different QTL, and gives the chromosome number, cM position and
effects of the QTL (assumed to be the same in each cross in which the
QTL is segregating).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error.prob</code></td>
<td>
<p>The genotyping error rate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing.prob</code></td>
<td>
<p>The rate of missing genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partial.missing.prob</code></td>
<td>
<p>When simulating an intercross or 4-way
cross, this gives the rate at which markers will be incompletely
informative (i.e., dominant or recessive).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.qtlgeno</code></td>
<td>
<p>If TRUE, genotypes for the simulated QTLs will be
included in the output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.errorind</code></td>
<td>
<p>If TRUE, and if <code>error.prob &gt; 0</code>, the
identity of genotyping errors will be included in the output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>Interference parameter; a non-negative integer.  0 corresponds
to no interference.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Probability that a chiasma comes from the no-interference
mechanism</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.function</code></td>
<td>
<p>Indicates whether to use the Haldane, Kosambi,
Carter-Falconer, or Morgan map function when converting genetic
distances into recombination fractions.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Meiosis is assumed to follow the Stahl model for crossover
interference (see the references, below), of which the no interference
model and the chi-square model are special cases.  Chiasmata on the
four-strand bundle are a superposition of chiasmata from two different
mechanisms.  With probability <code>p</code>, they arise by a mechanism
exhibiting no interference; the remainder come from a chi-square model
with inteference parameter <code>m</code>.  Note that <code>m=0</code> corresponds
to no interference, and with <code>p=0</code>, one gets a pure chi-square
model.
</p>
<p>QTLs are assumed to act additively, and the residual phenotypic
variation is assumed to be normally distributed with variance 1.
</p>
<p>The effect of a QTL is a pair of numbers,
(<code class="reqn">a,d</code>), where <code class="reqn">a</code> is the additive effect (half the difference
between the homozygotes) and <code class="reqn">d</code> is the dominance deviation (the
difference between the heterozygote and the midpoint between the
homozygotes).
</p>


<h3>Value</h3>

<p>A list with each component being an object of class <code>cross</code>.  See <code>read.cross</code> for
details.  The names (e.g. "AB", "AC", "BC") indicate the crosses.
</p>
<p>If <code>keep.qtlgeno</code> is TRUE, each cross object will contain a
component <code>qtlgeno</code> which is a matrix containing the QTL
genotypes (with complete data and no errors), coded as in the genotype
data.
</p>
<p>If <code>keep.errorind</code> is TRUE and errors were simulated, each
component of <code>geno</code> in each cross will each contain a matrix <code>errors</code>,
with 1's indicating simulated genotyping errors.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>References</h3>

<p>Broman, K. W., Kim, S., An\'e, C. and Payseur, B. A.  Mapping
quantitative trait loci to a phylogenetic tree.  In preparation.
</p>


<h3>See Also</h3>

 <p><code>scanPhyloQTL</code>,   <code>inferredpartitions</code>,
<code>summary.scanPhyloQTL</code>, <code>max.scanPhyloQTL</code>,
<code>plot.scanPhyloQTL</code>,
<code>sim.cross</code>, <code>read.cross</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# example map; drop X chromosome
data(map10)
map10 &lt;- map10[1:19]

# simulate data
x &lt;- simPhyloQTL(4, partition="AB|CD", crosses=c("AB", "AC", "AD"),
                 map=map10, n.ind=150,
                 model=c(1, 50, 0.5, 0))

# run calc.genoprob on each cross
x &lt;- lapply(x, calc.genoprob, step=2)

# scan genome, at each position trying all possible partitions
out &lt;- scanPhyloQTL(x, method="hk")

# maximum peak
max(out, format="lod")

# approximate posterior probabilities at peak
max(out, format="postprob")

# all peaks above a threshold for LOD(best) - LOD(2nd best)
summary(out, threshold=1, format="lod")

# all peaks above a threshold for LOD(best), showing approx post'r prob
summary(out, format="postprob", threshold=3)

# plot of results
plot(out)

## End(Not run)
</code></pre>


</div>