<div class="container">

<table style="width: 100%;"><tr>
<td>effectscan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot estimated QTL effects across the whole genome</h2>

<h3>Description</h3>

<p>This function is used to plot the estimated QTL effects along
selected chromosomes. For a backcross, there will be only one line,
representing the additive effect. For an intercross, there will be two
lines, representing the additive and dominance effects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">effectscan(cross, pheno.col=1, chr, get.se=FALSE, draw=TRUE,
           gap=25, ylim, mtick=c("line","triangle"),
           add.legend=TRUE, alternate.chrid=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>cross</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>Column number in the phenotype matrix which to be
drawn in the plot.  One may also give a character string matching
a phenotype name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes to be drawn in
the plot. This should be a vector of character strings referring to
chromosomes by name; numeric values are converted to strings.  Refer
to chromosomes with a preceding <code>-</code> to have all chromosomes but
those considered.  A logical (TRUE/FALSE) vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.se</code></td>
<td>
<p>If TRUE, estimated standard errors are calculated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>draw</code></td>
<td>
<p>If TRUE, draw the figure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>
<p>Gap separating chromosomes (in cM).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>
<p>Y-axis limits (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mtick</code></td>
<td>
<p>Tick mark type for markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.legend</code></td>
<td>
<p>If TRUE, add a legend.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternate.chrid</code></td>
<td>
<p>If TRUE and more than one chromosome is
plotted, alternate the placement of chromosome
axis labels, so that they may be more easily distinguished.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Passed to the function <code>plot</code> when it
is called.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The results of <code>sim.geno</code> are required for taking
account of missing genotype information.
</p>
<p>For a backcross, the additive effect is estimated as the difference
between the phenotypic averages for heterozygotes and homozygotes.
</p>
<p>For recombinant inbred lines, the additive effect is estimated as half
the difference between the phenotypic averages for the two homozygotes.
</p>
<p>For an intercross, the additive and dominance effects are estimated
from linear regression on <code class="reqn">a</code> and <code class="reqn">d</code> with <code class="reqn">a</code> = -1, 0, 1,
for the AA, AB and BB genotypes, respectively, and <code class="reqn">d</code> = 0, 1, 0,
for the AA, AB and BB genotypes, respectively.
</p>
<p>As usual, the X chromosome is a bit more complicated.  We estimate
separate additive effects for the two sexes, and for the two
directions within females.
</p>
<p>There is an internal function <code>plot.effectscan</code> that creates
the actual plot by calling <code>plot.scanone</code>.  In the case
<code>get.se=TRUE</code>, colored regions indicate <code class="reqn">\pm</code> 1 SE.
</p>


<h3>Value</h3>

<p>The results are returned silently, as an object of class
<code>"effectscan"</code>, which is the same as the form returned by the
function <code>scanone</code>, though with estimated effects
where LOD scores might be.  That is, it is a data frame with the first
two columns being chromosome ID and position (in cM), and subsequent
columns being estimated effects, and (if <code>get.se=TRUE</code>) standard
errors.
</p>


<h3>Author(s)</h3>

<p>Karl W. Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a></p>


<h3>References</h3>

<p>Sen, Ś. and Churchill, G. A. (2001) A statistical framework for quantitative
trait mapping.  <em>Genetics</em> <b>159</b>, 371–387.
</p>


<h3>See Also</h3>

 <p><code>effectplot</code>, <code>plotPXG</code>,
<code>sim.geno</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(fake.f2)

fake.f2 &lt;- sim.geno(fake.f2, step=2.5, n.draws=16)

# allelic effect on whole genome
effectscan(fake.f2)

# on chromosome 13, include standard errors
effectscan(fake.f2, chr="13", mtick="triangle", get.se=TRUE)
</code></pre>


</div>