<div class="container">

<table style="width: 100%;"><tr>
<td>calc.errorlod</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify likely genotyping errors</h2>

<h3>Description</h3>

<p>Calculates a LOD score for each genotype, measuring the evidence for
genotyping errors.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calc.errorlod(cross, error.prob=0.01,
              map.function=c("haldane","kosambi","c-f","morgan"),
              version=c("new","old"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>cross</code>.  See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error.prob</code></td>
<td>
<p>Assumed genotyping error rate used in the calculation
of the penetrance Pr(observed genotype | true genotype)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.function</code></td>
<td>
<p>Indicates whether to use the Haldane, Kosambi,
Carter-Falconer, or Morgan map function when converting genetic
distances into recombination fractions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>version</code></td>
<td>
<p>Specifies whether to use the original version of this
function or the current (preferred) version.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Calculates, for each individual at each marker, a LOD score
measuring the strength of evidence for a genotyping error, as
described by Lincoln and Lander (1992).
</p>
<p>In the latest version, evidence for a genotype being in
error is considered assuming that all other genotypes (for that
individual, on that chromosome) are correct.  The argument
<code>version</code> allows one to specify whether this new version is used,
or whether the original (old) version of the calculation is
performed.
</p>
<p>Note that values below 4 are generally not interesting.  Also note
that if markers are extremely tightly linked, <em>recombination
events</em> can give large error LOD scores.  The error LOD scores should
not be trusted blindly, but should be viewed as a tool for identifying
genotypes deserving further study.
</p>
<p>Use <code>top.errorlod</code> to print all genotypes with error
LOD scores above a specified threshold,
<code>plotErrorlod</code> to plot the error LOD scores for
specified chromosomes, and <code>plotGeno</code> to view the
observed genotype data with likely errors flagged.
</p>


<h3>Value</h3>

<p>The input <code>cross</code> object is returned with a component,
<code>errorlod</code>, added to each component of <code>cross$geno</code>.  The
<code>errorlod</code> component is a matrix of size (n.ind x n.mar).  An
attribute <code>"error.prob"</code> is set to the value of the corresponding
argument, for later reference.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>References</h3>

<p>Lincoln, S. E. and Lander, E. S. (1992) Systematic detection of
errors in genetic linkage data.  <em>Genomics</em> <b>14</b>, 604â€“610.
</p>


<h3>See Also</h3>

 <p><code>plotErrorlod</code>,
<code>top.errorlod</code>, <code>cleanGeno</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(hyper)

hyper &lt;- calc.errorlod(hyper,error.prob=0.01)

# print those above a specified cutoff
top.errorlod(hyper, cutoff=4)

# plot genotype data, flagging genotypes with error LOD &gt; cutoff
plotGeno(hyper, chr=1, ind=160:200, cutoff=7, min.sep=2)
</code></pre>


</div>