<div class="container">

<table style="width: 100%;"><tr>
<td>read.cross</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read data for a QTL experiment</h2>

<h3>Description</h3>

<p>Data for a QTL experiment is read from a set of files and converted
into an object of class <code>cross</code>.  The comma-delimited format
(<code>csv</code>) is recommended.  All formats require chromosome
assignments for the genetic markers, and assume that markers are in
their correct order.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.cross(format=c("csv", "csvr", "csvs", "csvsr", "mm", "qtx",
                    "qtlcart", "gary", "karl", "mapqtl", "tidy"),
           dir="", file, genfile, mapfile, phefile, chridfile,
           mnamesfile, pnamesfile, na.strings=c("-","NA"),
           genotypes=c("A","H","B","D","C"), alleles=c("A","B"),
           estimate.map=FALSE, convertXdata=TRUE, error.prob=0.0001,
           map.function=c("haldane", "kosambi", "c-f", "morgan"),
           BC.gen=0, F.gen=0, crosstype, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>Specifies the format of the data file or files. Details
on the various file formats are provided below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>Directory in which the data files will be found.  In
Windows, use forward slashes (<code>"/"</code>) or double backslashes
(<code>"\\"</code>) to specify directory trees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>The main input file for formats <code>csv</code>, <code>csvr</code>
and <code>mm</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genfile</code></td>
<td>
<p>File with genotype data (formats <code>csvs</code>,
<code>csvsr</code>, <code>karl</code>, <code>gary</code> and <code>mapqtl</code> only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapfile</code></td>
<td>
<p>File with marker position information (all
except the <code>csv</code> formats).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phefile</code></td>
<td>
<p>File with phenotype data (formats <code>csvs</code>,
<code>csvsr</code>, <code>karl</code>, <code>gary</code> and <code>mapqtl</code> only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chridfile</code></td>
<td>
<p>File with chromosome ID for each marker (<code>gary</code>
format only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mnamesfile</code></td>
<td>
<p>File with marker names (<code>gary</code> format only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pnamesfile</code></td>
<td>
<p>File with phenotype names (<code>gary</code> format
only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.strings</code></td>
<td>
<p>A vector of strings which are to be interpreted as
missing values (<code>csv</code> and <code>gary</code> formats only).  For the
<code>csv</code> formats, these are interpreted globally
for the entire
file, so missing value codes in phenotypes must not be valid
genotypes, and vice versa. For the <code>gary</code> format, these are
used only for the phenotype data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>A vector of character strings specifying the genotype
codes (<code>csv</code> formats only).  Generally this is a vector of
length 5, with the elements corresponding to AA, AB, BB, not BB
(i.e., AA or AB), and not AA (i.e., AB or BB).  <b>Note</b>: Pay
careful attention to the third and fourth of these; the order of
these can be confusing.
</p>
<p>If you are trying to read 4-way cross data, your file must have
genotypes coded as described below, and you need to set
<code>genotypes=NULL</code> so that no re-coding gets done.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>A vector of two one-letter character strings (or four,
for the four-way cross), to be used as labels for the two alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate.map</code></td>
<td>
<p>For all formats but <code>qtlcart</code>, <code>mapqtl</code>,
and <code>karl</code>: if TRUE and marker positions are not included in
the input files, the genetic map is estimated using the function
<code>est.map</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>convertXdata</code></td>
<td>
<p>If TRUE, any X chromosome genotype data is
converted to the internal standard, using columns <code>sex</code> and
<code>pgm</code> in the phenotype data if they available or by inference
if they are not.  If FALSE, the X chromsome data is read as is.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error.prob</code></td>
<td>
<p>In the case that the marker map must be estimated:
Assumed genotyping error rate used in the calculation of the
penetrance Pr(observed genotype | true genotype).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.function</code></td>
<td>
<p>In the case that the marker map must be estimated:
Indicates whether to use the Haldane, Kosambi,
Carter-Falconer, or Morgan map function when converting genetic
distances into recombination fractions. (Ignored if m &gt; 0.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BC.gen</code></td>
<td>
<p>Used only for cross type <code>"bcsft"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>F.gen</code></td>
<td>
<p>Used only for cross type <code>"bcsft"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crosstype</code></td>
<td>
<p>Optional character string to force a particular cross type.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments, passed to the function
<code>read.table</code> in the case of
<code>csv</code> and <code>csvr</code> formats.  In particular, one may use the
argument
<code>sep</code> to specify the field separator (the default is a comma),
<code>dec</code> to specify the character used for the decimal point
(the default is a period), and <code>comment.char</code> to specify a
character to indicate comment lines.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The available formats are comma-delimited (<code>csv</code>), rotated
comma-delimited (<code>csvr</code>), comma-delimited with separate files for
genotype and phenotype data (<code>csvs</code>), rotated comma-delimited
with separate files for genotype and phenotype data (<code>csvsr</code>),
Mapmaker (<code>mm</code>), Map Manager QTX (<code>qtx</code>), Gary Churchill's
format (<code>gary</code>), Karl Broman's format (<code>karl</code>) and
MapQTL/JoinMap (<code>mapqtl</code>).  The required files and their
specification for each format appears below. The comma-delimited
formats are recommended. Note that most of these formats work only
for backcross and intercross data.
</p>
<p>The <code>sampledata</code> directory in the package distribution contains
sample data files in multiple formats.    Also see
<a href="https://rqtl.org/sampledata/">https://rqtl.org/sampledata/</a>.
</p>
<p>The <code>...</code> argument enables additional arguments to be passed to
the function <code>read.table</code> in the case of <code>csv</code>
and <code>csvr</code> formats.  In particular, one may use the argument
<code>sep</code> to specify the field separator (the default is a comma),
<code>dec</code> to specify the character used for the decimal point (the
default is a period), and <code>comment.char</code> to specify a character
to indicate comment lines.
</p>


<h3>Value</h3>

<p>An object of class <code>cross</code>, which is a list with two components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>This is a list with elements corresponding to
chromosomes.  <code>names(geno)</code> contains the names of the
chromsomes.  Each chromosome is itself a list, and is given class
<code>A</code> or <code>X</code> according to whether it is autosomal
or the X chromosome.
</p>
<p>There are two components for each chromosome: <code>data</code>, a matrix
whose rows are individuals and whose columns are markers, and
<code>map</code>, either a vector of marker positions (in cM) or a matrix
of dim (<code>2 x n.mar</code>) where the rows correspond to marker
positions in female and male genetic distance, respectively.
</p>
<p>The genotype data gets converted into numeric codes, as follows.
</p>
<p>The genotype data for a backcross is coded as NA = missing,
1 = AA, 2 = AB.
</p>
<p>For an F2 intercross, the coding is NA = missing, 1 = AA, 2 = AB, 3
= BB, 4 = not BB (i.e. AA or AB; D in Mapmaker/qtl), 5 = not AA (i.e. AB
or BB; C in Mapmaker/qtl).
</p>
<p>For a 4-way cross, the mother and father are assumed to have
genotypes AB and CD, respectively.  The genotype data for the
progeny is assumed to be phase-known, with the following coding
scheme: NA = missing, 1 = AC, 2 = BC, 3 = AD, 4 = BD, 5 = A = AC or AD,
6 = B = BC or BD, 7 = C = AC or BC, 8 = D = AD or BD, 9 = AC or BD,
10 = AD or BC, 11 = not AC, 12 = not BC, 13 = not AD, 14 = not BD.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>data.frame of size (<code>n.ind x n.phe</code>) containing the
phenotypes.  If a phenotype with the name <code>id</code> or <code>ID</code> is
included, these identifiers will be used in <code>top.errorlod</code>,
<code>plotErrorlod</code>, and <code>plotGeno</code> as
identifiers for the individual.</p>
</td>
</tr>
</table>
<p>While the data format is complicated, there are a number of functions,
such as <code>subset.cross</code>, to
assist in pulling out portions of the data.
</p>


<h3>X chromosome</h3>

<p><b>The genotypes for the X chromosome require special care!</b>
</p>
<p>The X chromosome should be given chromosome identifier <code>X</code> or
<code>x</code>.  If it is labeled by a number or by <code>Xchr</code>, it will be
interpreted as an autosome.
</p>
<p>The phenotype data should contain a column named <code>"sex"</code> which
indicates the sex of each individual, either coded as <code>0</code>=female and
<code>1</code>=male, or as a factor with levels <code>female</code>/<code>male</code> or
<code>f</code>/<code>m</code>.  Case will be
ignored both in the name and in the factor levels.  If no such
phenotype column is included, it will be assumed that all individuals
are of the same sex.
</p>
<p>In the case of an intercross, the phenotype data may also contain a
column named <code>"pgm"</code> (for "paternal grandmother") indicating the
direction of the cross.  It should be coded as 0/1 with 0 indicating
the cross (AxB)x(AxB) or (BxA)x(AxB) and 1 indicating the cross
(AxB)x(BxA) or (BxA)x(BxA).  If no such phenotype column is included,
it will be assumed that all individuals come from the same direction
of cross.
</p>
<p>The internal storage of X chromosome data is quite different from that
of autosomal data.  Males are coded 1=AA and 2=BB; females with pgm==0
are coded 1=AA and 2=AB; and females with pgm==1 are coded 1=BB and
2=AB.  If the argument <code>convertXdata</code> is TRUE, conversion to this
format is made automatically; if FALSE, no conversion is done,
<code>summary.cross</code> will likely return a warning, and
most analyses will not work properly.
</p>
<p>Use of <code>convertXdata=FALSE</code> (in which case the X chromosome
genotypes will not be converted to our internal standard) can be
useful for diagnosing problems in the data, but will require some
serious mucking about in the internal data structure.
</p>


<h3>CSV format</h3>

<p>The input file is a comma-delimited text file.  A different field
separator may be specified via the argument <code>sep</code>, which will be passed
to the function <code>read.table</code>).  For example, in
Europe, it is common to use a comma in place of the decimal point in
numbers and so a semi-colon in place of a comma as the field
separator; such data may be read by using <code>sep=";"</code> and
<code>dec=","</code>.
</p>
<p>The first line should contain the phenotype names followed by the
marker names.  <b>At least one phenotype must be included</b>; for
example, include a numerical index for each individual.
</p>
<p>The second line should contain blanks in the phenotype columns,
followed by chromosome identifiers for each marker in all other
columns. If a chromosome has the identifier <code>X</code> or <code>x</code>, it
is assumed to be the X chromosome; otherwise, it is assumed to be an
autosome.
</p>
<p>An optional third line should contain blanks in the phenotype
columns, followed by marker positions, in cM.
</p>
<p>Marker order is taken from the cM positions, if provided; otherwise,
it is taken from the column order.
</p>
<p>Subsequent lines should give the data, with one line for each
individual, and with phenotypes followed by genotypes.  If possible,
phenotypes are made numeric; otherwise they are converted to factors.
</p>
<p>The genotype codes must be the same across all markers.  For example,
you can't have one marker coded AA/AB/BB and another coded A/H/B.
This includes genotypes for the X chromosome, for which hemizygous
individuals should be coded as if they were homoyzogous.
</p>
<p>The cross is determined to be a backcross if only the first two elements
of the <code>genotypes</code> string are found; otherwise, it is assumed to
be an intercross.
</p>


<h3>CSVr format</h3>

<p>This is just like the <code>csv</code> format, but rotated (or really
transposed), so that rows are columns and columns are rows.
</p>


<h3>CSVs format</h3>

<p>This is like the <code>csv</code> format, but with separate files for the
genotype and phenotype data.
</p>
<p>The first column in the genotype data must specify individuals'
identifiers, and there must be a column in the phenotype data with
precisely the same information (and with the same name).  These IDs
will be included in the data as a phenotype.  If the name <code>id</code> or
<code>ID</code> is used, these identifiers will be used in
<code>top.errorlod</code>, <code>plotErrorlod</code>, and
<code>plotGeno</code> as identifiers for the individual.
</p>
<p>The first row in each file contains the column names.  For the
phenotype file, these are the names of the phenotypes.  For the
genotype file, the first cell will be the name of the identifier
column (<code>id</code> or <code>ID</code>) and the subsequent fields will be the
marker names.
</p>
<p>In the genotype data file, the second row gives the chromosome IDs.
The cell in the second row, first column, must be blank.  A third
row giving cM positions of markers may be included, in which case the
cell in the third row, first column, must be blank.
</p>
<p>There need be no blank rows in the phenotype data file.
</p>


<h3>CSVsr format</h3>

<p>This is just like the <code>csvs</code> format, but with each file rotated
(or really transposed), so that rows are columns and columns are rows.
</p>


<h3>Mapmaker format</h3>

<p>This format requires two files.  The so-called rawfile, specified by
the argument <code>file</code>, contains the genotype and phenotype
data. Rows beginning with the symbol <code>#</code> are ignored.  The first
line should be either <code>data type f2 intercross</code> or
<code>data type f2 backcross</code>.  The second line should begin with
three numbers indicating the numbers of individuals, markers and
phenotypes in the file.  This line may include the word <code>symbols</code>
followed by symbol assignments (see the documentation for mapmaker,
and cross your fingers).  The rest of the lines give genotype data
followed by phenotype data, with marker and phenotype names always
beginning with the <code>*</code> symbol.
</p>
<p>A second file contains the genetic map information, specified with
the argument <code>mapfile</code>.  The map file may be in
one of two formats.  The function will determine which format of map
file is presented.
</p>
<p>The simplest format for the map file is not standard for the Mapmaker
software, but is easy to create.  The file contains two or three
columns separated by white space and with no header row.  The first
column gives the chromosome assignments.  The second column gives the
marker names, with markers listed in the order along the chromosomes.
An optional third column lists the map positions of the markers.
</p>
<p>Another possible format for the map file is the <code>.maps</code>
format, which is produced by Mapmaker.  The code for reading this
format was written by Brian Yandell.
</p>
<p>Marker order is taken from the map file, either by the order they are
presented or by the cM positions, if specified.
</p>


<h3>Map Manager QTX format</h3>

<p>This format requires a single file (that produced by the Map Manager
QTX program).
</p>


<h3>QTL Cartographer format</h3>

<p>This format requires two files: the <code>.cro</code> and <code>.map</code> files
for QTL Cartographer (produced by the QTL Cartographer
sub-program, Rmap and Rcross).
</p>
<p>Note that the QTL Cartographer cross types are converted as follows:
RF1 to riself, RF2 to risib, RF0 (doubled haploids) to bc, B1 or B2 to
bc, RF2 or SF2 to f2.
</p>


<h3>Tidy format</h3>

<p>This format requires three simple CSV files, separating the genotype,
phenotype, and marker map information so that each file may be of a
simple form.
</p>


<h3>Gary format</h3>

<p>This format requires the six files.  All files have default names, and
so the file names need not be specified if the default names are used.
</p>
<p><code>genfile</code> (default = <code>"geno.dat"</code>) contains the genotype
data.  The file contains one line per individual, with genotypes for
the set of markers separated by white space.  Missing values are
coded as 9, and genotypes are coded as 0/1/2 for AA/AB/BB.
</p>
<p><code>mapfile</code> (default = <code>"markerpos.txt"</code>) contains two
columns with no header row: the marker names in the first column and
their cM position in the second column.  If marker positions are not
available, use <code>mapfile=NULL</code>, and a dummy map will be inserted.
</p>
<p><code>phefile</code> (default = <code>"pheno.dat"</code>) contains the phenotype
data, with one row for each mouse and one column for each phenotype.
There should be no header row, and missing values are coded as
<code>"-"</code>.
</p>
<p><code>chridfile</code> (default = <code>"chrid.dat"</code>) contains the
chromosome identifier for each marker.
</p>
<p><code>mnamesfile</code> (default = <code>"mnames.txt"</code>) contains the marker
names.
</p>
<p><code>pnamesfile</code> (default = <code>"pnames.txt"</code>) contains the names
of the phenotypes.  If phenotype names file is not available, use
<code>pnamesfile=NULL</code>; arbitrary phenotype names will then be
assigned.
</p>


<h3>Karl format</h3>

<p>This format requires three files; all files have default names, and so
need not be specified if the default name is used.
</p>
<p><code>genfile</code> (default = <code>"gen.txt"</code>) contains the genotype
data.  The file contains one line per individual, with genotypes
separated by white space.  Missing values are coded 0; genotypes are
coded as 1/2/3/4/5 for AA/AB/BB/not BB/not AA.
</p>
<p><code>mapfile</code> (default = <code>"map.txt"</code>) contains the map
information, in the following complicated format: <br><br><code>n.chr</code> <br><code>n.mar(1) rf(1,1) rf(1,2) ... rf(1,n.mar(1)-1)</code><br><code>mar.name(1,1)</code><br><code>mar.name(1,2)</code><br><code>...</code><br><code>mar.name(1,n.mar(1))</code><br><code>n.mar(2)</code><br><code>...</code><br><code>etc.</code> <br></p>
<p><code>phefile</code> (default = <code>"phe.txt"</code>) contains a matrix of
phenotypes, with one individual per line.  The first line in the
file should give the phenotype names.
</p>


<h3>MapQTL format</h3>

<p>This format requires three files, described in the manual of the
MapQTL program (same as JoinMap).
</p>
<p><code>genfile</code> corresponds to the loc file containing the genotype
data. Each marker and its genotypes should be on a single line.
</p>
<p><code>mapfile</code> corresponds to the map file containing the linkage
group assignment, marker names and their map positions.
</p>
<p><code>phefile</code> corresponds to the qua file containing the phenotypes.
</p>
<p>For the moment, only 4-way crosses are supported (CP population type
in MapQTL).
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>; Brian
S. Yandell; Aaron Wolen</p>


<h3>References</h3>

<p>Broman, K. W. and Sen,
Ś. (2009) <em>A
guide to QTL mapping with R/qtl.</em>  Springer.  <a href="https://rqtl.org/book/">https://rqtl.org/book/</a>
</p>


<h3>See Also</h3>

 <p><code>subset.cross</code>, <code>summary.cross</code>,
<code>plot.cross</code>, <code>c.cross</code>, <code>clean.cross</code>,
<code>write.cross</code>, <code>sim.cross</code>, <code>read.table</code>.
The <code>sampledata</code> directory in the package distribution contains
sample data files in multiple formats.  Also see
<a href="https://rqtl.org/sampledata/">https://rqtl.org/sampledata/</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# CSV format
dat1 &lt;- read.cross("csv", dir="Mydata", file="mydata.csv")

# CSVS format
dat2 &lt;- read.cross("csvs", dir="Mydata", genfile="mydata_gen.csv",
                   phefile="mydata_phe.csv")

# you can read files directly from the internet
datweb &lt;- read.cross("csv", "https://rqtl.org/sampledata",
                     "listeria.csv")

# Mapmaker format
dat3 &lt;- read.cross("mm", dir="Mydata", file="mydata.raw",
                   mapfile="mydata.map")

# Map Manager QTX format
dat4 &lt;- read.cross("qtx", dir="Mydata", file="mydata.qtx")

# QTL Cartographer format
dat5 &lt;- read.cross("qtlcart", dir="Mydata", file="qtlcart.cro",
                   mapfile="qtlcart.map")

# Gary format
dat6 &lt;- read.cross("gary", dir="Mydata", genfile="geno.dat",
                   mapfile="markerpos.txt", phefile="pheno.dat",
                   chridfile="chrid.dat", mnamesfile="mnames.txt",
                   pnamesfile="pnames.txt")

# Karl format
dat7 &lt;- read.cross("karl", dir="Mydata", genfile="gen.txt",
                   phefile="phe.txt", mapfile="map.txt")
## End(Not run)
</code></pre>


</div>