<div class="container">

<table style="width: 100%;"><tr>
<td>plot.scanPhyloQTL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot LOD curves from single-QTL scan to map QTL to a phylogenetic tree</h2>

<h3>Description</h3>

<p>Plot the LOD curves for each partition for a genome scan with a single
diallelic QTL (the
output of <code>scanPhyloQTL</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'scanPhyloQTL'
plot(x, chr, incl.markers=TRUE,
     col, xlim, ylim, lwd=2, gap=25, mtick=c("line", "triangle"),
     show.marker.names=FALSE, alternate.chrid=FALSE, legend=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>"scanPhyloQTL"</code>, as output by
<code>scanPhyloQTL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes to plot.
This should be a vector of character strings referring to chromosomes
by name; numeric values are converted to strings.  Refer to
chromosomes with a preceding <code>-</code> to have all chromosomes but
those considered.  A logical (TRUE/FALSE) vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>incl.markers</code></td>
<td>
<p>Indicate whether to plot line segments at the
marker locations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>Optional vector of colors to use for each partition.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlim</code></td>
<td>
<p>Limits for x-axis (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>
<p>Limits for y-axis (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd</code></td>
<td>
<p>Line width.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>
<p>Gap separating chromosomes (in cM).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mtick</code></td>
<td>
<p>Tick mark type for markers (line segments or
upward-pointing triangels).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.marker.names</code></td>
<td>
<p>If TRUE, show the marker names along the x axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternate.chrid</code></td>
<td>
<p>If TRUE and more than one chromosome is
plotted, alternate the placement of chromosome
axis labels, so that they may be more easily distinguished.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>Indicates whether to include a legend in the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Passed to the function <code>plot.scanone</code> when it
is called.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>None.</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>References</h3>

<p>Broman, K. W., Kim, S., An\'e, C. and Payseur, B. A.  Mapping
quantitative trait loci to a phylogenetic tree.  In preparation.
</p>


<h3>See Also</h3>

 <p><code>scanPhyloQTL</code>, <code>max.scanPhyloQTL</code>,
<code>summary.scanPhyloQTL</code>, <code>plot.scanone</code>,
<code>inferredpartitions</code>,
<code>simPhyloQTL</code>,
<code>par</code>, <code>colors</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# example map; drop X chromosome
data(map10)
map10 &lt;- map10[1:19]

# simulate data
x &lt;- simPhyloQTL(4, partition="AB|CD", crosses=c("AB", "AC", "AD"),
                 map=map10, n.ind=150,
                 model=c(1, 50, 0.5, 0))

# run calc.genoprob on each cross
x &lt;- lapply(x, calc.genoprob, step=2)

# scan genome, at each position trying all possible partitions
out &lt;- scanPhyloQTL(x, method="hk")

# maximum peak
max(out, format="lod")

# approximate posterior probabilities at peak
max(out, format="postprob")

# all peaks above a threshold for LOD(best) - LOD(2nd best)
summary(out, threshold=1, format="lod")

# all peaks above a threshold for LOD(best), showing approx post'r prob
summary(out, format="postprob", threshold=3)

# plot of results
plot(out)

## End(Not run)
</code></pre>


</div>