<div class="container">

<table style="width: 100%;"><tr>
<td>simFounderSnps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate founder SNPs for a multiple-strain RIL</h2>

<h3>Description</h3>

<p>Simulate genotype data for the founding strains for a panel of
multiple-strain RIL.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simFounderSnps(map, n.str=c("4","8"), pat.freq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>A list whose components are vectors containing the marker
locations on each of the chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.str</code></td>
<td>
<p>Number of founding strains (4 or 8).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pat.freq</code></td>
<td>
<p>Frequency of SNP genotype patterns in the founder (a
vector of length <code>n.str</code>/2 + 1): (monoallelic, SNP unique to one
founder, SNP present in 2 founders, [and, for the case of 8 founders,
SNP in 3/8 founders, SNP in 4/8 founders].)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The SNPs are simulated to be in linkage equilibrium.
</p>


<h3>Value</h3>

<p>A vector of the same length as there are chromosomes in <code>map</code>,
with each component being a matrix of 0's and 1's, of dim <code>n.str</code>
x <code>n.mar</code>.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>See Also</h3>

 <p><code>sim.map</code>, <code>sim.cross</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(map10)
x &lt;- simFounderSnps(map10, "8", c(0, 0.5, 0.2, 0.2, 0.1))
</code></pre>


</div>