<div class="container">

<table style="width: 100%;"><tr>
<td>scantwopermhk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Permutation test for 2d genome scan by Haley-Knott regression</h2>

<h3>Description</h3>

<p>Perform a permutation test with a two-dimensional genome scan with a
two-QTL model, with possible allowance for additive covariates, by
Haley-Knott regression.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scantwopermhk(cross, chr, pheno.col=1,
              addcovar=NULL, weights=NULL, n.perm=1,
              batchsize=1000,
              perm.strata=NULL, perm.Xsp=NULL,
              verbose=FALSE, assumeCondIndep=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>cross</code>. See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes for which LOD
scores should be calculated.  This should be a vector of character
strings referring to chromosomes by name; numeric values are
converted to strings.  Refer to chromosomes with a preceding <code>-</code>
to have all chromosomes but those considered.  A logical (TRUE/FALSE)
vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>Column number in the phenotype matrix which should be
used as the phenotype.  This should be a single value (numeric index
or character string for a phenotype name), but it may also be a
vector of numeric values with length equal to the number
of individuals in the cross, in which case it is taken to be a vector
of individuals' phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>Additive covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>Optional weights of individuals.  Should be either NULL
or a vector of length n.ind containing positive weights. Used only
in the case <code>model="normal"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>Number of permutation replicates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batchsize</code></td>
<td>
<p>If <code>n.perm &gt; batchsize</code>, permutations will be
run in batches of no more than <code>batchsize</code> permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.strata</code></td>
<td>
<p>Used to perform a
stratified permutation test.  This should be a vector with the same
number of individuals as in the cross data.  Unique values indicate
the individual strata, and permutations will be performed within the
strata.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.Xsp</code></td>
<td>
<p>If TRUE, run separate permutations for A:A, A:X, and
X:X. In this case, <code>n.perm</code> refers to the number of permutations
for the A:A part; more permutations are used for the A:X and X:X parts,
as estimates of quantiles farther out into the tails are needed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE, display information about the progress of
calculations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assumeCondIndep</code></td>
<td>
<p>If TRUE, assume conditional independence of QTL
genotypes given marker genotypes.  This is an approximation, but it
may speed things up.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is a scaled-back version of the permutation test provided by
<code>scantwo</code>: only for a normal model with Haley-Knott
regression, and not allowing interactive covariates.
</p>
<p>This is an attempt to speed things up and attentuate the memory usage
problems in <code>scantwo</code>.
</p>
<p>In the case of <code>perm.Xsp=TRUE</code> (X-chr-specific thresholds), we
use a stratified permutation test, stratified by sex and
cross-direction.
</p>


<h3>Value</h3>

<p>A list with six
different LOD scores from each of the permutation replicates.
First, the maximum LOD score for the full model (two QTLs plus an
interaction).  Second, for each pair of
chromosomes, we take the difference between the full LOD and the
maximum single-QTL LOD for those two chromosomes, and then maximize
this across chromosome pairs.  Third, for each pair of chromosomes we
take the difference between the maximum full LOD and the maximum
additive LOD, and then maximize this across chromosome pairs.  Fourth,
the maximum LOD score for the additive QTL model.  Fifth, for each
pair of chromosomes, we take the difference between the additive LOD
and the maximum single-QTL LOD for those two chromosomes, and then
maximize this across chromosome pairs.  Finally, the maximum
single-QTL LOD score (that is, from a single-QTL scan).  The latter is
not used in <code>summary.scantwoperm</code>, but does get
calculated at each permutation, so we include it for the sake of
completeness.
</p>
<p>If <code>perm.Xsp=TRUE</code>, this is a list of lists, for the A:A, A:X,
and X:X sections, each being a list as described above.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>; Hao Wu</p>


<h3>References</h3>

<p>Churchill, G. A. and Doerge, R. W. (1994) Empirical threshold values for
quantitative trait mapping.  <em>Genetics</em> <b>138</b>, 963–971.
</p>
<p>Haley, C. S. and Knott, S. A. (1992) A simple regression method for mapping
quantitative trait loci in line crosses using flanking markers.
<em>Heredity</em> <b>69</b>, 315–324.
</p>


<h3>See Also</h3>

 <p><code>scantwo</code>, <code>plot.scantwoperm</code>,
<code>summary.scantwoperm</code>,
<code>c.scantwoperm</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(fake.f2)

fake.f2 &lt;- calc.genoprob(fake.f2, step=5)
operm &lt;- scantwopermhk(fake.f2, n.perm=2)
summary(operm, alpha=0.05)
</code></pre>


</div>