<div class="container">

<table style="width: 100%;"><tr>
<td>scanPhyloQTL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Single-QTL genome scan to map QTL to a phylogenetic tree</h2>

<h3>Description</h3>

<p>Jointly consider multiple intercrosses with a single diallelic QTL
model, considering all possible partitions of the strains into the two
QTL allele groups.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scanPhyloQTL(crosses, partitions, chr, pheno.col=1,
             model=c("normal", "binary"), method=c("em", "imp", "hk"),
             addcovar, maxit=4000, tol=0.0001, useAllCrosses=TRUE,
             verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>crosses</code></td>
<td>
<p>A list with each component being an intercross, as an object of class
<code>cross</code> (see <code>read.cross</code> for details).  The names
(of the form "AB") indicate the strains in the cross.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partitions</code></td>
<td>
<p>A vector of character strings of the form "AB|CD" or
"A|BCD" indicating the set of paritions of the strains into two
allele groups.  If missing, all partitions should be considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes for which LOD
scores should be calculated.  This should be a vector of character
strings referring to chromosomes by name; numeric values are
converted to strings.  Refer to chromosomes with a preceding <code>-</code>
to have all chromosomes but those considered.  A logical (TRUE/FALSE)
vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>Column number in the phenotype matrix which should be
used as the phenotype.  This can be a vector of integers; for methods
<code>"hk"</code> and <code>"imp"</code> this can be considerably faster than doing
them one at a time.  One may also give a character strings matching
the phenotype names.  Finally, one may give a numeric vector of
phenotypes, in which case it must have the length equal to the number
of individuals in the cross, and there must be either non-integers or
values &lt; 1 or &gt; no. phenotypes; this last case may be useful for studying
transformations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>The phenotype model: the usual normal model or a model for
binary traits</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Indicates whether to use the EM algorithm,
imputation, or Haley-Knott regression.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>Optional set of additive covariates to include in the
analysis, as a list with the same length as <code>crosses</code>.  They
must be numeric vectors or matrices, as for <code>scanone</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxit</code></td>
<td>
<p>Maximum number of iterations for method <code>"em"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Tolerance value for determining convergence for method
<code>"em"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useAllCrosses</code></td>
<td>
<p>If TRUE, use all crosses in the analysis of all
partitions, with crosses not segregating the QTL included in the
estimation of the residual variance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE, print information about progress.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The aim is to jointly consider multiple intercrosses to not just map
QTL but to also, under the assumption of a single diallelic QTL,
identify the set of strains with each QTL allele.
</p>
<p>For each partition (of the strains into two groups) that is under
consideration, we pull out the set of crosses that are segregating the
QTL, re-code the alleles, and combine the crosses into one large cross.
Crosses not segregating the QTL are also used, though with no QTL
effects.
</p>
<p>Additive covariate indicators for the crosses are included in the
analysis, to allow for the possibility that there are overall shifts
in the phenotypes between crosses.
</p>


<h3>Value</h3>

<p>A data frame, as for the output of <code>scanone</code>, though with
LOD score columns for each partition that is considered.  The result
is given class <code>"scanPhyloQTL"</code>.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a></p>


<h3>References</h3>

<p>Broman, K. W., Kim, S., An\'e, C. and Payseur, B. A.  Mapping
quantitative trait loci to a phylogenetic tree.  In preparation.
</p>


<h3>See Also</h3>

 <p><code>plot.scanPhyloQTL</code>,
<code>summary.scanPhyloQTL</code>, <code>max.scanPhyloQTL</code>,
<code>inferredpartitions</code>,
<code>simPhyloQTL</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
# example map; drop X chromosome
data(map10)
map10 &lt;- map10[1:19]


# simulate data
x &lt;- simPhyloQTL(4, partition="AB|CD", crosses=c("AB", "AC", "AD"),
                 map=map10, n.ind=150,
                 model=c(1, 50, 0.5, 0))

# run calc.genoprob on each cross
## Not run: x &lt;- lapply(x, calc.genoprob, step=2)


# scan genome, at each position trying all possible partitions
out &lt;- scanPhyloQTL(x, method="hk")

# maximum peak
max(out, format="lod")

# approximate posterior probabilities at peak
max(out, format="postprob")

# all peaks above a threshold for LOD(best) - LOD(2nd best)
summary(out, threshold=1, format="lod")

# all peaks above a threshold for LOD(best), showing approx post'r prob
summary(out, format="postprob", threshold=3)

# plot results
plot(out)
</code></pre>


</div>