<div class="container">

<table style="width: 100%;"><tr>
<td>plotErrorlod</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot grid of error LOD values</h2>

<h3>Description</h3>

<p>Plot a grid of the LOD scores indicating which genotypes are
likely to be in error.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotErrorlod(x, chr, ind, breaks=c(-Inf,2,3,4.5,Inf),
              col=c("white","gray85","hotpink","purple3"),
              alternate.chrid=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>cross</code>.  See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes to be drawn in
the plot. This should be a vector of character strings referring to
chromosomes by name; numeric values are converted to strings.  Refer
to chromosomes with a preceding <code>-</code> to have all chromosomes but
those considered.  A logical (TRUE/FALSE) vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind</code></td>
<td>
<p>Indicates the individuals for which the error LOD scores
should be plotted (passed to <code>subset.cross</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>breaks</code></td>
<td>
<p>A set of breakpoints for the colors; must give one more
breakpoint than color.  Intervals are open on the left and closed on
the right, except for the lowest interval.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>A vector of colors to appear in the image.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternate.chrid</code></td>
<td>
<p>If TRUE and more than one chromosome is
plotted, alternate the placement of chromosome
axis labels, so that they may be more easily distinguished.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Ignored at this point.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Uses <code>image</code> to plot a grid with different shades
of pixels to indicate which genotypes are likely to be in error.
</p>
<p>Darker pixels have higher error LOD scores:
<code class="reqn">LOD \le 2</code> in white;
<code class="reqn">2 &lt; LOD \le 3</code> in gray;
<code class="reqn">3 &lt; LOD \le 4.5</code> in pink;
<code class="reqn">LOD &gt; 4.5</code> in purple.
</p>


<h3>Value</h3>

<p>None.</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>References</h3>

<p>Lincoln, S. E. and Lander, E. S. (1992) Systematic detection of
errors in genetic linkage data.  <em>Genomics</em> <b>14</b>, 604â€“610.
</p>


<h3>See Also</h3>

 <p><code>calc.errorlod</code>,
<code>top.errorlod</code>, <code>image</code>,
<code>subset.cross</code>, <code>plotGeno</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(hyper)

# Calculate error LOD scores
hyper &lt;- calc.errorlod(hyper,error.prob=0.01)

# plot the error LOD scores; print those above a specified cutoff
plotErrorlod(hyper)
plotErrorlod(hyper,chr=1)
</code></pre>


</div>