<div class="container">

<table style="width: 100%;"><tr>
<td>cim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Composite interval mapping</h2>

<h3>Description</h3>

<p>Composite interval mapping by a scheme from QTL Cartographer:
forward selection at the markers (here, with filled-in genotype data)
to a fixed number, followed by interval mapping with the selected
markers as covariates, dropping marker covariates if they are within
some fixed window size of the location under test.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cim(cross, pheno.col=1, n.marcovar=3, window=10,
    method=c("em", "imp", "hk", "ehk"),
    imp.method=c("imp", "argmax"), error.prob=0.0001,
    map.function=c("haldane", "kosambi", "c-v", "morgan"),
    addcovar=NULL, n.perm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>cross</code>. See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>Column number in the phenotype matrix which should be
used as the phenotype.  One may also give a character string matching
a phenotype name. Finally, one may give a numeric vector of
phenotypes, in which case it must have the length equal to the number
of individuals in the cross, and there must be either non-integers or
values &lt; 1 or &gt; no. phenotypes; this last case may be useful for studying
transformations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.marcovar</code></td>
<td>
<p>Number of marker covariates to use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window</code></td>
<td>
<p>Window size, in cM.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Indicates whether to use the EM algorithm,
imputation, Haley-Knott regression, or the extended Haley-Knott
method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imp.method</code></td>
<td>
<p>Method used to impute any missing marker genotype
data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error.prob</code></td>
<td>
<p>Genotyping error probability assumed when imputing
the missing marker genotype data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.function</code></td>
<td>
<p>Map function used when imputing the missing marker
genotype data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>Optional numeric matrix of additional covariates to include.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>If specified, a permutation test is performed rather than
an analysis of the observed data.  This argument defines the number
of permutation replicates.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We first use <code>fill.geno</code> to impute any missing marker
genotype data, either via a simple random imputation or using the
Viterbi algorithm.
</p>
<p>We then perform forward selection to a fixed number of markers.  These
will be used (again, with any missing data filled in) as covariates in
the subsequent genome scan.
</p>


<h3>Value</h3>

<p>The function returns an object of the same form as the function
<code>scanone</code>:
</p>
<p>If <code>n.perm</code> is missing, the function returns the scan results as
a data.frame with three columns: chromosome, position, LOD score.
Attributes indicate the names and positions of the chosen marker
covariates.
</p>
<p>If <code>n.perm</code> &gt; 0, the function results the results of a
permutation test: a vector giving the genome-wide maximum LOD score in
each of the permutations.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a></p>


<h3>References</h3>

<p>Jansen, R. C. (1993) Interval mapping of multiple quantitative trait
loci.  <em>Genetics</em>, <b>135</b>, 205–211.
</p>
<p>Jansen, R. C. and Stam, P. (1994) High resolution of quantitative
traits into multiple loci via interval mapping.  <em>Genetics</em>,
<b>136</b>, 1447-1455.
</p>
<p>Zeng, Z. B. (1993) Theoretical basis for separation of multiple linked
gene effects in mapping quantitative trait loci.
<em>Proc. Natl. Acad. Sci. USA</em>, <b>90</b>, 10972–10976.
</p>
<p>Zeng, Z. B. (1994) Precision mapping of quantitative trait loci.
<em>Genetics</em>, <b>136</b>, 1457–1468.
</p>


<h3>See Also</h3>

 <p><code>add.cim.covar</code>, <code>scanone</code>,
<code>summary.scanone</code>, <code>plot.scanone</code>,
<code>fill.geno</code></p>


<h3>Examples</h3>

<pre><code class="language-R">data(hyper)
hyper &lt;- calc.genoprob(hyper, step=2.5)


out &lt;- scanone(hyper)
out.cim &lt;- cim(hyper, n.marcovar=3)
plot(out, out.cim, chr=c(1,4,6,15), col=c("blue", "red"))

add.cim.covar(out.cim, chr=c(1,4,6,15))
</code></pre>


</div>