<div class="container">

<table style="width: 100%;"><tr>
<td>plotGeno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot observed genotypes, flagging likely errors</h2>

<h3>Description</h3>

<p>Plot the genotypes on a particular chromosome for a set of
individuals, flagging likely errors.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotGeno(x, chr, ind, include.xo=TRUE, horizontal=TRUE,
          cutoff=4, min.sep=2, cex=1.2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>cross</code>.  See
<code>read.cross</code> for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>The chromosome to plot.  Only one chromosome is
allowed. (This should be a character string referring to the
chromosomes by name.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind</code></td>
<td>
<p>Vector of individuals to plot (passed to <code>subset.cross</code>).  If missing, all individuals
are plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include.xo</code></td>
<td>
<p>If TRUE, plot X's in intervals having a
crossover. Not available for a 4-way cross.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>horizontal</code></td>
<td>
<p>If TRUE, chromosomes are plotted horizontally.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p>Genotypes with error LOD scores above this value are
flagged as possible errors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.sep</code></td>
<td>
<p>Markers separated by less than this value (as a
percent of the chromosome length) are pulled apart, so that they may
be distinguished in the picture.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>Character expansion for the size of points in the plot.
Larger numbers give larger points; see <code>par</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Ignored at this point.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Plots the genotypes for a set of individuals.  Likely errors are
indicated by red squares.  In a backcross, genotypes AA and AB are
indicated by white and black circles, respectively.  In an intercross,
genotypes AA, AB and BB are indicated by white, gray, and black
circles, respectively, and the partially missing genotypes "not BB"
(D in mapmaker) and "not AA" (C in mapmaker) are indicated by green
and orange circles, respectively.
</p>
<p>For the X chromosome in a backcross or intercross, hemizygous males
are plotted as if they were homozygous (that is, with white and black
circles).
</p>
<p>For a 4-way cross, two lines are plotted for each individual.  The
left or upper line indicates the allele A (white) or B (black); the
right or lower line indicates the allele C (white) or D (black).  For
the case that genotype is known to be only AC/BD or AD/BC, we use
green and orange, respectively.
</p>


<h3>Value</h3>

<p>None.</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>See Also</h3>

 <p><code>calc.errorlod</code>,
<code>top.errorlod</code>, <code>subset.cross</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(hyper)

# Calculate error LOD scores
hyper &lt;- calc.errorlod(hyper,error.prob=0.01)

# print those above a specified cutoff
top.errorlod(hyper,cutoff=4)

# plot genotype data, flagging genotypes with error LOD &gt; cutoff
plotGeno(hyper, chr=1, ind=160:200, cutoff=7, min.sep=2)
</code></pre>


</div>