<div class="container">

<table style="width: 100%;"><tr>
<td>MQM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Introduction to Multiple QTL Model (MQM) mapping</h2>

<h3>Description</h3>

<p>Overview of the MQM mapping functions
</p>


<h3>Introduction</h3>


<p>Multiple QTL Mapping (MQM) provides a sensitive approach for
mapping quantititive trait loci (QTL) in experimental populations. MQM
adds higher statistical power compared to many other methods.  The
theoretical framework of MQM was introduced and explored by
Ritsert Jansen, explained in the ‘Handbook of Statistical Genetics’
(see references), and used effectively in practical research, with the
commercial ‘mapqtl’ software package. Here we present the first free
and open source implementation of MQM, with extra features like high
performance parallelization on multi-CPU computers, new plots and
significance testing.
</p>
<p>MQM is an automatic three-stage procedure in which, in the first
stage, missing data is ‘augmented’. In other words, rather than guessing one
likely genotype, multiple
genotypes are modeled with their estimated
probabilities.  In the second stage important markers are selected by
multiple regression and backward elimination. In the third stage a QTL is moved
along the chromosomes using these pre-selected markers as cofactors,
except for the markers in the window around the interval under study. QTL are
(interval) mapped using the most ‘informative’ model through maximum
likelihood.  A refined and automated procedure for cases with large
numbers of marker cofactors is included.  The method internally
controls false discovery rates (FDR) and lets users test different QTL
models by elimination of non-significant cofactors.
</p>


<p>R/qtl-MQM has the following advantages:
</p>

<ul>
<li>
<p> Higher power to detect linked as well as unlinked QTL, as long as the QTL explain a reasonable amount of variation
</p>
</li>
<li>
<p> Protection against overfitting, because it fixes the residual variance from the full model. For this reason more parameters (cofactors) can be used compared to, for example, CIM
</p>
</li>
<li>
<p> Prevention of ghost QTL (between two QTL in coupling phase)
</p>
</li>
<li>
<p> Detection of negating QTL (QTL in repulsion phase)</p>
</li>
</ul>
<h3>Note</h3>


<p>The current implementation of R/qtl-MQM has the following
limitations: (1) MQM is limited to experimental crosses F2,
BC, and selfed RIL, (2) MQM does not treat sex
chromosomes differently from autosomal chromosomes - though one can
introduce sex as a cofactor. Future versions of R/qtl-MQM may improve on
these points. Check the website and change log
(<a href="https://github.com/kbroman/qtl/blob/main/NEWS.md">https://github.com/kbroman/qtl/blob/main/NEWS.md</a>) for updates.
</p>



<h3>Author(s)</h3>

<p>Ritsert C Jansen; Danny Arends; Pjotr Prins; Karl W Broman <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> 
</p>


<h3>References</h3>



<ul>
<li>
<p> Arends D, Prins P, Jansen RC.  R/qtl: High-throughput multiple QTL mapping. <em>Bioinformatics</em>, to appear
</p>
</li>
<li>
<p> Jansen RC, (2007) Quantitative trait loci in inbred lines. Chapter 18 of <em>Handbook of Stat. Genetics</em> 3rd edition. John Wiley &amp; Sons, Ltd.
</p>
</li>
<li>
<p> Jansen RC, Nap JP (2001), Genetical genomics: the added value from segregation. <em>Trends in Genetics</em>, <b>17</b>, 388–391.
</p>
</li>
<li>
<p> Jansen RC, Stam P (1994), High resolution of quantitative traits into multiple loci via interval mapping. <em>Genetics</em>, <b>136</b>, 1447–1455.
</p>
</li>
<li>
<p> Jansen RC (1993), Interval mapping of multiple quantitative trait loci. <em>Genetics</em>, <b>135</b>, 205–211.
</p>
</li>
<li>
<p> Swertz MA, Jansen RC. (2007), Beyond standardization: dynamic software infrastructures for systems biology. <em>Nat Rev Genet.</em> <b>3</b>, 235–243.
</p>
</li>
<li>
<p> Dempster, A. P., Laird, N. M. and Rubin, D. B. (1977) Maximum likelihood from incomplete data via the EM algorithm.  <em>J. Roy. Statist. Soc.</em> B, <b>39</b>, 1–38.
</p>

</li>
</ul>
<h3>See Also</h3>



<ul>
<li>
<p> The MQM tutorial: <a href="https://rqtl.org/tutorials/MQM-tour.pdf">https://rqtl.org/tutorials/MQM-tour.pdf</a>
</p>
</li>
<li> <p><code>MQM</code> - MQM description and references
</p>
</li>
<li> <p><code>mqmscan</code> - Main MQM single trait analysis
</p>
</li>
<li> <p><code>mqmscanall</code> - Parallellized traits analysis
</p>
</li>
<li> <p><code>mqmaugment</code> - Augmentation routine for estimating missing data
</p>
</li>
<li> <p><code>mqmautocofactors</code> - Set cofactors using marker density
</p>
</li>
<li> <p><code>mqmsetcofactors</code> - Set cofactors at fixed locations
</p>
</li>
<li> <p><code>mqmpermutation</code> - Estimate significance levels
</p>
</li>
<li> <p><code>scanone</code> - Single QTL scanning

</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
data(map10)                    # Genetic map modeled after mouse

# simulate a cross (autosomes 1-10)
qtl &lt;- c(3,15,1,0)             # QTL model: chr, pos'n, add've &amp; dom effects
cross &lt;- sim.cross(map10[1:10],qtl,n=100,missing.prob=0.01)

# MQM
crossaug &lt;- mqmaugment(cross)  # Augmentation
cat(crossaug$mqm$Nind,'real individuals retained in dataset',
    crossaug$mqm$Naug,'individuals augmented\n')

result &lt;- mqmscan(crossaug)    # Scan

# show LOD interval of the QTL on chr 3
lodint(result,chr=3)

</code></pre>


</div>