<div class="container">

<table style="width: 100%;"><tr>
<td>scantwo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Two-dimensional genome scan with a two-QTL model</h2>

<h3>Description</h3>

<p>Perform a two-dimensional genome scan with a two-QTL model, with
possible allowance for covariates.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scantwo(cross, chr, pheno.col=1, model=c("normal","binary"),
        method=c("em","imp","hk","mr","mr-imp","mr-argmax"),
        addcovar=NULL, intcovar=NULL, weights=NULL,
        use=c("all.obs", "complete.obs"),
        incl.markers=FALSE, clean.output=FALSE,
        clean.nmar=1, clean.distance=0,
        maxit=4000, tol=1e-4,
        verbose=TRUE, n.perm, perm.Xsp=FALSE, perm.strata=NULL,
        assumeCondIndep=FALSE, batchsize=250, n.cluster=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>cross</code>. See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes for which LOD
scores should be calculated.  This should be a vector of character
strings referring to chromosomes by name; numeric values are
converted to strings.  Refer to chromosomes with a preceding <code>-</code>
to have all chromosomes but those considered.  A logical (TRUE/FALSE)
vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>Column number in the phenotype matrix which should be
used as the phenotype.  This can be a vector of integers; for methods
<code>"hk"</code> and <code>"imp"</code> this can be considerably faster than doing
them one at a time.  One may also give character strings matching
the phenotype names.  Finally, one may give a numeric vector of
phenotypes, in which case it must have the length equal to the number
of individuals in the cross, and there must be either non-integers or
values &lt; 1 or &gt; no. phenotypes; this last case may be useful for studying
transformations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>The phenotype model: the usual normal model or a model for
binary traits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Indicates whether to use the
the EM algorithm, imputation, Haley-Knott regression, or marker
regression.  Marker regression is performed either by dropping
individuals with missing genotypes (<code>"mr"</code>), or by first filling
in missing data using a single imputation (<code>"mr-imp"</code>) or by the
Viterbi algorithm (<code>"mr-argmax"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>Additive covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intcovar</code></td>
<td>
<p>Interactive covariates (interact with QTL genotype).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>Optional weights of individuals.  Should be either NULL
or a vector of length n.ind containing positive weights. Used only
in the case <code>model="normal"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use</code></td>
<td>
<p>In the case that multiple phenotypes are selected to be
scanned, this argument indicates whether to use all individuals,
including those missing some phenotypes, or just those individuals
that have data on all selected phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>incl.markers</code></td>
<td>
<p>If FALSE, do calculations only at points on an
evenly spaced grid.  If <code>calc.genoprob</code> or
<code>sim.geno</code> were run with
<code>stepwidth="variable"</code> or <code>stepwidth="max"</code>, we force <code>incl.markers=TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clean.output</code></td>
<td>
<p>If TRUE, clean the output with
<code>clean.scantwo</code>, replacing LOD scores for pairs of
positions that are not well separated with 0.  In permutations, this
will be done for each permutation replicate.  This can be important
for the case of <code>method="em"</code>, as there can be difficulty with
algorithm convergence in these regions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clean.nmar</code></td>
<td>
<p>If <code>clean.output=TRUE</code>, this is the number of
markers that must separate two positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clean.distance</code></td>
<td>
<p>If <code>clean.output=TRUE</code>, this is the cM distance
that must separate two positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxit</code></td>
<td>
<p>Maximum number of iterations; used
only with method <code>"em"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Tolerance value for determining convergence; used only with
method <code>"em"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE, display information about the progress of
calculations.  For method <code>"em"</code>, if <code>verbose</code> is an integer
above 1, further details on the progress of the algorithm will be
displayed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>If specified, a permutation test is performed rather than
an analysis of the observed data.  This argument defines the number
of permutation replicates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.Xsp</code></td>
<td>
<p>If <code>n.perm</code> &gt; 0, so that a permutation test will
be performed, this indicates whether separate permutations should be
performed for the autosomes and the X chromosome, in order to get an
X-chromosome-specific LOD threshold.  In this case, additional
permutations are performed for the X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.strata</code></td>
<td>
<p>If <code>n.perm</code> &gt; 0, this may be used to perform a
stratified permutation test.  This should be a vector with the same
number of individuals as in the cross data.  Unique values indicate
the individual strata, and permutations will be performed within the
strata.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assumeCondIndep</code></td>
<td>
<p>If TRUE, assume conditional independence of QTL
genotypes given marker genotypes.  This is an approximation, but it
may speed things up.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batchsize</code></td>
<td>
<p>The number of phenotypes (or permutations) to be run
as a batch; used only for methods <code>"hk"</code> and <code>"imp"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cluster</code></td>
<td>
<p>If the package <code>snow</code> is available and
<code>n.perm</code> &gt; 0, permutations are run in parallel using this number
of nodes.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Standard interval mapping (<code>method="em"</code>) and Haley-Knott
regression (<code>method="hk"</code>) require that multipoint genotype probabilities are
first calculated using <code>calc.genoprob</code>.  The
imputation method uses the results of <code>sim.geno</code>.
</p>
<p>The method <code>"em"</code> is standard interval mapping by the EM algorithm
(Dempster et al. 1977; Lander and Botstein 1989).  Marker regression
(<code>method="mr"</code>) is simply linear regression of phenotypes on
marker genotypes  (individuals with missing genotypes are
discarded). Haley-Knott regression (<code>method="hk"</code>) uses the
regression of phenotypes on multipoint genotype probabilities.  The
imputation method (<code>method="imp"</code>) uses the pseudomarker
algorithm described by Sen and Churchill (2001).
</p>
<p>Individuals with missing phenotypes are dropped.
</p>
<p>In the presence of covariates, the full model is
</p>
<p style="text-align: center;"><code class="reqn">y = \mu + \beta_{q_1} + \beta_{q_2} + \beta_{q_1 \times q_2} +
    A \gamma + Z \delta_{q_1} + Z \delta_{q_2} + Z \delta_{q_1 \times q_2} +
    \epsilon</code>
</p>

<p>where <code class="reqn">q_1</code> and <code class="reqn">q_2</code> are the unknown QTL genotypes at two
locations, <em>A</em> is a matrix of covariates, and <em>Z</em> is a
matrix of covariates that interact with QTL genotypes.  The columns of
<em>Z</em> are forced to be contained in the matrix <em>A</em>.
</p>
<p>The above full model is compared to the additive QTL model,
</p>
<p style="text-align: center;"><code class="reqn">y = \mu + \beta_{q_1} + \beta_{q_2} + A \gamma + Z \delta_{q_1} + Z
    \delta_{q_2} + \epsilon</code>
</p>

<p>and also to the null model, with no QTL,
</p>
<p style="text-align: center;"><code class="reqn">y = \mu + A \gamma + \epsilon</code>
</p>

<p>In the case that <code>n.perm</code> is specified, the R function
<code>scantwo</code> is called repeatedly.
</p>
<p>For <code>model="binary"</code>, a logistic regression model is used.
</p>


<h3>Value</h3>

<p>If <code>n.perm</code> is missing, the function returns a list with class
<code>"scantwo"</code> and containing three components.  The first component
is a matrix of dimension [tot.pos x tot.pos]; the upper triangle
contains the LOD scores for the additive model, and the lower triangle
contains the LOD scores for the full model.  The diagonal contains the
results of <code>scanone</code>. The second component of the
output is a data.frame indicating the locations at which the two-QTL
LOD scores were calculated.  The first column is the chromosome
identifier, the second column is the position in cM, the third column
is a 1/0 indicator for ease in later pulling out only the equally
spaced positions, and the fourth column indicates whether the position
is on the X chromosome or not.  The final component is a version of
the results of <code>scanone</code> including sex and/or cross
direction as additive covariates, which is needed for a proper
calculation of conditional LOD scores.
</p>
<p>If <code>n.perm</code> is specified, the function returns a list with six
different LOD scores from each of the permutation replicates.
First, the maximum LOD score for the full model (two QTLs plus an
interaction).  Second, for each pair of
chromosomes, we take the difference between the full LOD and the
maximum single-QTL LOD for those two chromosomes, and then maximize
this across chromosome pairs.  Third, for each pair of chromosomes we
take the difference between the maximum full LOD and the maximum
additive LOD, and then maximize this across chromosome pairs.  Fourth,
the maximum LOD score for the additive QTL model.  Fifth, for each
pair of chromosomes, we take the difference between the additive LOD
and the maximum single-QTL LOD for those two chromosomes, and then
maximize this across chromosome pairs.  Finally, the maximum
single-QTL LOD score (that is, from a single-QTL scan).  The latter is
not used in <code>summary.scantwo</code>, but does get
calculated at each permutation, so we include it for the sake of
completeness.
</p>
<p>If <code>n.perm</code> is specified and <code>perm.Xsp=TRUE</code>, the result is
a list with the permutation results for the regions A:A, A:X, and X:X,
each of which is a list with the six different LOD scores. Independent
permutations are performed in each region, <code>n.perm</code> is the number
of permutations for the A:A region; additional permutations are are
used for the A:X and X:X parts, as estimates of quantiles farther out
into the tails are needed.
</p>


<h3>X chromosome</h3>

<p>The X chromosome must be treated specially in QTL mapping.
</p>
<p>As in <code>scanone</code>, if both males and females are
included, male hemizygotes are allowed to be different from female
homozygotes, and the null hypothesis must be changed in order to ensure
that sex- or pgm-differences in the phenotype do not results in spurious
linkage to the X chromosome.  (See the help file for
<code>scanone</code>.)
</p>
<p>If <code>n.perm</code> is specified and <code>perm.Xsp=TRUE</code>,
X-chromosome-specific permutations are performed, to obtain separate
thresholds for the regions A:A, A:X, and X:X.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>; Hao Wu</p>


<h3>References</h3>

<p>Churchill, G. A. and Doerge, R. W. (1994) Empirical threshold values for
quantitative trait mapping.  <em>Genetics</em> <b>138</b>, 963–971.
</p>
<p>Dempster, A. P., Laird, N. M. and Rubin, D. B. (1977) Maximum
likelihood from incomplete data via the EM algorithm.  <em>J. Roy.
Statist. Soc.</em> B, <b>39</b>, 1–38.
</p>
<p>Haley, C. S. and Knott, S. A. (1992) A simple regression method for mapping
quantitative trait loci in line crosses using flanking markers.
<em>Heredity</em> <b>69</b>, 315–324.
</p>
<p>Lander, E. S. and Botstein, D. (1989) Mapping Mendelian factors underlying
quantitative traits using RFLP linkage maps.  <em>Genetics</em>
<b>121</b>, 185–199.
</p>
<p>Sen, Ś. and Churchill, G. A. (2001) A statistical framework for quantitative
trait mapping.  <em>Genetics</em> <b>159</b>, 371–387.
</p>
<p>Soller, M., Brody, T. and Genizi, A. (1976) On the power of experimental
designs for the detection of linkage between marker loci and
quantitative loci in crosses between inbred lines.
<em>Theor. Appl. Genet.</em> <b>47</b>, 35–39.
</p>


<h3>See Also</h3>

 <p><code>plot.scantwo</code>, <code>summary.scantwo</code>,
<code>scanone</code>, <code>max.scantwo</code>,
<code>summary.scantwoperm</code>,
<code>c.scantwoperm</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(fake.f2)

fake.f2 &lt;- calc.genoprob(fake.f2, step=5)
out.2dim &lt;- scantwo(fake.f2, method="hk")
plot(out.2dim)

# permutations

## Not run: permo.2dim &lt;- scantwo(fake.f2, method="hk", n.perm=1000)
summary(permo.2dim, alpha=0.05)

# summary with p-values
summary(out.2dim, perms=permo.2dim, pvalues=TRUE,
        alphas=c(0.05, 0.10, 0.10, 0.05, 0.10))

# covariates
data(fake.bc)

fake.bc &lt;- calc.genoprob(fake.bc, step=10)

ac &lt;- pull.pheno(fake.bc, c("sex","age"))
ic &lt;- pull.pheno(fake.bc, "sex")

out &lt;- scantwo(fake.bc, method="hk", pheno.col=1,
               addcovar=ac, intcovar=ic)
plot(out)
</code></pre>


</div>