<div class="container">

<table style="width: 100%;"><tr>
<td>plotInfo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot the proportion of missing genotype information</h2>

<h3>Description</h3>

<p>Plot a measure of the proportion of missing information in the
genotype data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotInfo(x, chr, method=c("entropy","variance","both"), step=1,
          off.end=0, error.prob=0.001,
          map.function=c("haldane","kosambi","c-f","morgan"),
          alternate.chrid=FALSE, fourwaycross=c("all", "AB", "CD"),
          include.genofreq=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>cross</code>.  See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes to plot.
This should be a vector of character strings referring to chromosomes
by name; numeric values are converted to strings.  Refer to
chromosomes with a preceding <code>-</code> to have all chromosomes but
those considered.  A logical (TRUE/FALSE) vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Indicates whether to plot the entropy version of the
information, the variance version, or both.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>step</code></td>
<td>
<p>Maximum distance (in cM) between positions at which the
missing information is calculated, though for <code>step=0</code>,
it is are calculated only at the marker locations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>off.end</code></td>
<td>
<p>Distance (in cM) past the terminal markers on each
chromosome to which the genotype probability calculations will be
carried.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error.prob</code></td>
<td>
<p>Assumed genotyping error rate used in the calculation
of the penetrance Pr(observed genotype | true genotype).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.function</code></td>
<td>
<p>Indicates whether to use the Haldane, Kosambi or
Carter-Falconer map function when converting genetic distances into
recombination fractions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternate.chrid</code></td>
<td>
<p>If TRUE and more than one chromosome is
plotted, alternate the placement of chromosome
axis labels, so that they may be more easily distinguished.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fourwaycross</code></td>
<td>
<p>For a phase-known four-way cross, measure missing
genotype information overall (<code>"all"</code>), or just for the alleles
from the first parent (<code>"AB"</code>) or from the second parent (<code>"CD"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include.genofreq</code></td>
<td>
<p>If TRUE, estimated genotype frequencies (from
the results of
<code>calc.genoprob</code> averaged across the individuals) are
included as additional columns in the output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Passed to <code>plot.scanone</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The entropy version of the missing information: for a single
individual at a single genomic position, we measure the missing
information as <code class="reqn">H = \sum_g p_g \log p_g / \log n</code>, where <code class="reqn">p_g</code> is the probability of the
genotype <code class="reqn">g</code>, and <code class="reqn">n</code> is the number of possible genotypes,
defining <code class="reqn">0 \log 0 = 0</code>.  This takes values between 0
and 1, assuming the value 1 when the genotypes (given the marker data)
are equally likely and 0 when the genotypes are completely determined.
We calculate the missing information at a particular position as the
average of <code class="reqn">H</code> across individuals.  For an intercross, we don't
scale by <code class="reqn">\log n</code> but by the entropy in the case of genotype
probabilities (1/4, 1/2, 1/4).
</p>
<p>The variance version of the missing information: we calculate the
average, across individuals, of the variance of the genotype
distribution (conditional on the observed marker data) at a particular
locus, and scale by the maximum such variance.
</p>
<p>Calculations are done in C (for the sake of speed in the presence of
little thought about programming efficiency) and the plot is created
by a call to <code>plot.scanone</code>.
</p>
<p>Note that <code>summary.scanone</code> may be used to display
the maximum missing information on each chromosome.
</p>


<h3>Value</h3>

<p>An object with class <code>scanone</code>: a data.frame with columns the
chromosome IDs and cM positions followed by the entropy and/or
variance version of the missing information.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> </p>


<h3>See Also</h3>

 <p><code>plot.scanone</code>,
<code>plotMissing</code>, <code>calc.genoprob</code>,
<code>geno.table</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(hyper)

plotInfo(hyper,chr=c(1,4))

# save the results and view maximum missing info on each chr
info &lt;- plotInfo(hyper)
summary(info)

plotInfo(hyper, bandcol="gray70")
</code></pre>


</div>