<div class="container">

<table style="width: 100%;"><tr>
<td>summary.scanPhyloQTL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Summarize the results a genome scan to map a QTL to a phylogenetic tree</h2>

<h3>Description</h3>

<p>Print the maximum LOD scores for each partition on each chromosome,
from the results of <code>scanPhyloQTL</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'scanPhyloQTL'
summary(object, format=c("postprob", "lod"),
        threshold, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An object output by the function
<code>scanPhyloQTL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>Indicates whether to provide LOD scores or approximate
posterior probabilities; see Details below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>A threshold determining which chromosomes should be
output; see Details below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Ignored at this point.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is used to report chromosomes deemed interesting from a one-QTL
genome scan to map QTL to a phylogenetic tree (by <code>scanPhyloQTL</code>).
</p>
<p>For <code>format="lod"</code>, the output contains the maximum LOD score for
each partition on each chromosome (which do not necessarily occur at
the same position).  The position corresponds to the peak location for
the partition with the largest LOD score on that chromosome.  The
last column is the overall maximum LOD (across partitions) on that
chromosome.  The second-to-last column is the inferred partition
(i.e., that with the largest LOD
score.  The third-to-last column is the difference between the LOD score for
the best partition and that for the second-best.
</p>
<p>For <code>format="postprob"</code>, the final column contains the maximum
LOD score across partitions.  But instead of providing the LOD
scores for each partition, these are converted to approximate
posterior probabilities under the assumption of a single diallelic QTL
on that chromosome:  on each chromosome, we take
<code class="reqn">10^{LOD}</code> for the partitions and rescale them to sum to 1.
</p>
<p>The <code>threshold</code> argument is applied to the last column (the
maximum LOD score across partitions).
</p>


<h3>Value</h3>

<p>An object of class <code>summary.scanPhyloQTL</code>, to be printed by
<code>print.summary.scanPhyloQTL</code>.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a></p>


<h3>References</h3>

<p>Broman, K. W., Kim, S., An\'e, C. and Payseur, B. A.  Mapping
quantitative trait loci to a phylogenetic tree.  In preparation.
</p>


<h3>See Also</h3>

 <p><code>scanPhyloQTL</code>, <code>plot.scanPhyloQTL</code>,
<code>max.scanPhyloQTL</code>, <code>summary.scanone</code>,
<code>inferredpartitions</code>,
<code>simPhyloQTL</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# example map; drop X chromosome
data(map10)
map10 &lt;- map10[1:19]

# simulate data
x &lt;- simPhyloQTL(4, partition="AB|CD", crosses=c("AB", "AC", "AD"),
                 map=map10, n.ind=150,
                 model=c(1, 50, 0.5, 0))

# run calc.genoprob on each cross
x &lt;- lapply(x, calc.genoprob, step=2)

# scan genome, at each position trying all possible partitions
out &lt;- scanPhyloQTL(x, method="hk")

# maximum peak
max(out, format="lod")

# approximate posterior probabilities at peak
max(out, format="postprob")

# all peaks above a threshold for LOD(best) - LOD(2nd best)
summary(out, threshold=1, format="lod")

# all peaks above a threshold for LOD(best), showing approx post'r prob
summary(out, format="postprob", threshold=3)

# plot of results
plot(out)

## End(Not run)
</code></pre>


</div>