<div class="container">

<table style="width: 100%;"><tr>
<td>mqmaugment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MQM augmentation</h2>

<h3>Description</h3>

<p>Fill in missing genotypes for MQM mapping. For each missing or incomplete
marker it fills in (or ‘augments’) all possible genotypes, thus creating new
candidate ‘individuals’. The probability of each indidual is calculated using
information on neighbouring markers and recombination frequencies. When a
genotype of an augmented genotype is less likely than the <code>minprob</code>
parameter it is dropped from the dataset. The <em>augmented</em> list of
individuals is returned in a new cross object. For a full discussion on
augmentation see the MQM tutorial online.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mqmaugment(cross, maxaugind=82, minprob=0.1,
           strategy=c("default","impute","drop"),
           verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>

<p>An object of class <code>cross</code>. See <code>read.cross</code> for details. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxaugind</code></td>
<td>

<p>Maximum number of augmentations per individual.  The default of 82
allows for six missing markers for an individual in a BC cross
(<code class="reqn">2^6=64</code>) and four missing markers in an F2 (<code class="reqn">3^4=81</code>). When a
large number of markers are missing this default number is quickly
reached.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minprob</code></td>
<td>

<p>Return individuals with augmented genotypes that have at least this probability of
occurring. <code>minprob</code> is a value between 0 and 1. For example a value of 0.5
will drop all genotypes that are half as likely as the most likely
genotype (candidate of the individual). The default value of 0.1 will drop
all genotypes that are less likely of ocurring than 1 in 10, compared
against the most likely genotype. Use a value of 1.0 to return a single
filled in genotype for each individual.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strategy</code></td>
<td>

<p>When individuals have too much missing data and augmentation fails three
options are provided:
1. <code>"default"</code>: Calculate genotypes at missing marker positions,
accounting for <code>minprob</code>, and add this individual to the set.
2. <code>"impute"</code>: Calculate the most likely genotypes at missing marker
positions and impute <code>maxaugind</code> individual-variants around the most
likely genotype.
3. <code>"drop"</code>: Drop individuals that cannot be augmented from the
dataset, this option is not advised because information from the dropped
individuals will be lost.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> If TRUE, give verbose output </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns the cross object with augmented individuals (many individuals from
the data set will be repeated multiple times). Some individuals may have been
dropped completely when the probability falls below <code>minprob</code>. An added
component to the cross object named <code>mqm</code> contains information on
exactly which individuals are retained and repeated.
</p>


<h3>Note</h3>

<p>The sex chromosome 'X' is treated like autosomes during augmentation.
With an F2 the sex chromosome is not considered. This will change in
a future version of MQM.
Run with <code>verbose=TRUE</code> to verify how many individuals are augmented
versus moved to the second augmentation round. This could have an effect
on the resulting dataset or check the return <code>cross$mqm</code> values. Compare
results by using <code>minprob=1</code>.
</p>


<h3>Author(s)</h3>

<p>Ritsert C Jansen; Danny Arends; Pjotr Prins; Karl W Broman <a href="mailto:broman@wisc.edu">broman@wisc.edu</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>fill.geno</code> - Alternative routine for estimating missing data

</p>
</li>
<li>
<p> The MQM tutorial: <a href="https://rqtl.org/tutorials/MQM-tour.pdf">https://rqtl.org/tutorials/MQM-tour.pdf</a>
</p>
</li>
<li> <p><code>MQM</code> - MQM description and references
</p>
</li>
<li> <p><code>mqmscan</code> - Main MQM single trait analysis
</p>
</li>
<li> <p><code>mqmscanall</code> - Parallellized traits analysis
</p>
</li>
<li> <p><code>mqmaugment</code> - Augmentation routine for estimating missing data
</p>
</li>
<li> <p><code>mqmautocofactors</code> - Set cofactors using marker density
</p>
</li>
<li> <p><code>mqmsetcofactors</code> - Set cofactors at fixed locations
</p>
</li>
<li> <p><code>mqmpermutation</code> - Estimate significance levels
</p>
</li>
<li> <p><code>scanone</code> - Single QTL scanning

</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
data(map10)                    # Genetic map modeled after mouse

# simulate a cross (autosomes 1-10)
qtl &lt;- c(3,15,1,0)             # QTL model: chr, pos'n, add've &amp; dom effects
cross &lt;- sim.cross(map10[1:10],qtl,n=100,missing.prob=0.01)

# MQM
crossaug &lt;- mqmaugment(cross)  # Augmentation
cat(crossaug$mqm$Nind,'real individuals retained in dataset',
    crossaug$mqm$Naug,'individuals augmented\n')

result &lt;- mqmscan(crossaug)    # Scan

# show LOD interval of the QTL on chr 3
lodint(result,chr=3)

</code></pre>


</div>