<div class="container">

<table style="width: 100%;"><tr>
<td>scanone</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genome scan with a single QTL model</h2>

<h3>Description</h3>

<p>Genome scan with a single QTL model, with possible allowance for
covariates, using any of several possible models for the phenotype and
any of several possible numerical methods.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scanone(cross, chr, pheno.col=1, model=c("normal","binary","2part","np"),
        method=c("em","imp","hk","ehk","mr","mr-imp","mr-argmax"),
        addcovar=NULL, intcovar=NULL, weights=NULL,
        use=c("all.obs", "complete.obs"), upper=FALSE,
        ties.random=FALSE, start=NULL, maxit=4000,
        tol=1e-4, n.perm, perm.Xsp=FALSE, perm.strata=NULL, verbose,
        batchsize=250, n.cluster=1, ind.noqtl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>An object of class <code>cross</code>. See
<code>read.cross</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Optional vector indicating the chromosomes for which LOD
scores should be calculated.  This should be a vector of character
strings referring to chromosomes by name; numeric values are
converted to strings.  Refer to chromosomes with a preceding <code>-</code>
to have all chromosomes but those considered.  A logical (TRUE/FALSE)
vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>Column number in the phenotype matrix which should be
used as the phenotype.  This can be a vector of integers; for methods
<code>"hk"</code> and <code>"imp"</code> this can be considerably faster than doing
them one at a time.  One may also give a character strings matching
the phenotype names.  Finally, one may give a numeric vector of
phenotypes, in which case it must have the length equal to the number
of individuals in the cross, and there must be either non-integers or
values &lt; 1 or &gt; no. phenotypes; this last case may be useful for studying
transformations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>The phenotype model: the usual normal model, a model for
binary traits, a two-part model or non-parametric analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Indicates whether to use the EM algorithm,
imputation, Haley-Knott regression, the extended Haley-Knott method,
or marker regression.  Not all methods are available for all models.
Marker regression is performed either by dropping individuals with
missing genotypes (<code>"mr"</code>), or by first filling in missing data
using a single imputation (<code>"mr-imp"</code>) or by the Viterbi
algorithm (<code>"mr-argmax"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addcovar</code></td>
<td>
<p>Additive covariates;
allowed only for the normal and binary models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intcovar</code></td>
<td>
<p>Interactive covariates (interact with QTL genotype);
allowed only for the normal and binary models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>Optional weights of individuals.  Should be either NULL
or a vector of length n.ind containing positive weights.  Used only
in the case <code>model="normal"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use</code></td>
<td>
<p>In the case that multiple phenotypes are selected to be
scanned, this argument indicates whether to use all individuals,
including those missing some phenotypes, or just those individuals
that have data on all selected phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upper</code></td>
<td>
<p>Used only for the two-part model; if true, the
"undefined" phenotype is the maximum observed phenotype; otherwise,
it is the smallest observed phenotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ties.random</code></td>
<td>
<p>Used only for the non-parametric "model"; if TRUE,
ties in the phenotypes are ranked at random.  If FALSE, average ranks
are used and a corrected LOD score is calculated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>Used only for the EM algorithm with the normal model and
no covariates.  If <code>NULL</code>, use the usual starting values; if
length 1, use random initial weights for EM; otherwise, this should
be a vector of length n+1 (where n is the number of possible
genotypes for the cross), giving the initial values for EM.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxit</code></td>
<td>
<p>Maximum number of iterations for methods <code>"em"</code> and
<code>"ehk"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Tolerance value for determining convergence for methods
<code>"em"</code> and <code>"ehk"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>If specified, a permutation test is performed rather than
an analysis of the observed data.  This argument defines the number
of permutation replicates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.Xsp</code></td>
<td>
<p>If <code>n.perm</code> &gt; 0, so that a permutation test will
be performed, this indicates whether separate permutations should be
performed for the autosomes and the X chromosome, in order to get an
X-chromosome-specific LOD threshold.  In this case, additional
permutations are performed for the X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm.strata</code></td>
<td>
<p>If <code>n.perm</code> &gt; 0, this may be used to perform a
stratified permutation test.  This should be a vector with the same
number of individuals as in the cross data.  Unique values indicate
the individual strata, and permutations will be performed within the
strata.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>In the case <code>n.perm</code> is specified, display
information about the progress of the permutation tests.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batchsize</code></td>
<td>
<p>The number of phenotypes (or permutations) to be run
as a batch; used only for methods <code>"hk"</code> and <code>"imp"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cluster</code></td>
<td>
<p>If the package <code>snow</code> is available and
<code>n.perm</code> &gt; 0, permutations are run in parallel using this number
of nodes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.noqtl</code></td>
<td>
<p>Indicates individuals who should not be allowed a QTL
effect (used rarely, if at all); this is a logical vector of same
length as there are individuals in the cross.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Use of the EM algorithm, Haley-Knott regression, and the extended
Haley-Knott method require that multipoint genotype probabilities are
first calculated using <code>calc.genoprob</code>.  The
imputation method uses the results of <code>sim.geno</code>.
</p>
<p>Individuals with missing phenotypes are dropped.
</p>
<p>In the case that <code>n.perm</code>&gt;0, so that a permutation
test is performed, the R function <code>scanone</code> is called repeatedly.
If <code>perm.Xsp=TRUE</code>, separate permutations are performed for the
autosomes and the X chromosome, so that an X-chromosome-specific
threshold may be calculated.  In this case, <code>n.perm</code> specifies
the number of permutations used for the autosomes; for the X
chromosome, <code>n.perm</code> <code class="reqn">\times \, L_A/L_X</code> permutations
will be run, where <code class="reqn">L_A</code> and <code class="reqn">L_X</code> are the total genetic
lengths of the autosomes and X chromosome, respectively.  More
permutations are needed for the X chromosome in order to obtain
thresholds of similar accuracy.
</p>
<p>For further details on the models, the methods and the use of
covariates, see below.
</p>


<h3>Value</h3>

<p>If <code>n.perm</code> is missing, the function returns a data.frame whose
first two columns contain the chromosome IDs and cM positions.
Subsequent columns contain the LOD scores for each phenotype.
In the case of the two-part model, there are three LOD score columns
for each phenotype: LOD(<code class="reqn">p,\mu</code>), LOD(<code class="reqn">p</code>) and
LOD(<code class="reqn">\mu</code>).   The result is given class <code>"scanone"</code> and
has attributes  <code>"model"</code>, <code>"method"</code>, and
<code>"type"</code> (the latter is the type of cross analyzed).
</p>
<p>If <code>n.perm</code> is specified, the function returns the results of a
permutation test and the output has class <code>"scanoneperm"</code>.  If
<code>perm.Xsp=FALSE</code>, the function returns a matrix with
<code>n.perm</code> rows, each row containing the genome-wide
maximum LOD score for each of the phenotypes.  In the case of the
two-part model, there are three columns for each phenotype,
corresponding to the three different LOD scores. If
<code>perm.Xsp=TRUE</code>, the result contains separate permutation results
for the autosomes and the X chromosome respectively, and an attribute
indicates the lengths of the chromosomes and an indicator of which
chromosome is X.
</p>


<h3>Models</h3>

<p><b>The normal model</b> is the standard model for QTL mapping (see
Lander and Botstein 1989).  The
residual phenotypic variation is assumed to follow a normal
distribution, and analysis is analogous to analysis of variance.
</p>
<p><b>The binary model</b> is for the case of a binary phenotype, which
must have values 0 and 1.  The proportions of 1's in the different
genotype groups are compared.  Currently only methods <code>em</code>, <code>hk</code>, and
<code>mr</code> are available for this model.  See Xu and Atchley (1996) and
Broman (2003).
</p>
<p><b>The two-part model</b> is appropriate for the case of a spike in
the phenotype distribution (for example, metastatic density when many
individuals show no metastasis, or survival time following an
infection when individuals may recover from the infection and fail to
die).  The two-part model was described by
Boyartchuk et al. (2001) and Broman (2003).  Individuals with QTL
genotype <code class="reqn">g</code> have probability <code class="reqn">p_g</code> of having an
undefined phenotype (the spike), while if their phenotype is defined,
it comes from a normal distribution with mean <code class="reqn">\mu_g</code> and
common standard deviation <code class="reqn">\sigma</code>. Three LOD scores are
calculated: LOD(<code class="reqn">p,\mu</code>) is for the test of the hypothesis
that <code class="reqn">p_g = p</code> and <code class="reqn">\mu_g = \mu</code>.
LOD(<code class="reqn">p</code>) is for the test that <code class="reqn">p_g = p</code> while the
<code class="reqn">\mu_g</code> may vary. LOD(<code class="reqn">\mu</code>) is for the test that
<code class="reqn">\mu_g = \mu</code> while the <code class="reqn">p_g</code> may vary.
</p>
<p><b>With the non-parametric "model"</b>, an extension of the
Kruskal-Wallis test is used; this is similar to the method described
by Kruglyak and Lander (1995).  In the case of incomplete genotype
information (such as at locations between genetic markers), the
Kruskal-Wallis statistic is modified so that the rank for each
individual is weighted by the genotype probabilities, analogous to
Haley-Knott regression.  For this method, if the argument
<code>ties.random</code> is TRUE, ties in the phenotypes are assigned random
ranks; if it is FALSE, average ranks are used and a corrected LOD
score is calculate.  Currently the <code>method</code> argument is ignored
for this model.
</p>


<h3>Methods</h3>

<p><b><code>em</code></b>: maximum likelihood is performed via the
EM algorithm (Dempster et al. 1977), first used in this context by
Lander and Botstein (1989).
</p>
<p><b><code>imp</code></b>: multiple imputation is used, as described by Sen
and Churchill (2001).
</p>
<p><b><code>hk</code></b>: Haley-Knott regression is used (regression of the
phenotypes on the multipoint QTL genotype probabilities), as described
by Haley and Knott (1992).
</p>
<p><b><code>ehk</code></b>: the extended Haley-Knott method is used (like H-K,
but taking account of the variances), as described in Feenstra et
al. (2006).
</p>
<p><b><code>mr</code></b>: Marker regression is used.  Analysis is performed
only at the genetic markers, and individuals with missing genotypes
are discarded.  See Soller et al. (1976).
</p>


<h3>Covariates</h3>

<p>Covariates are allowed only for the normal and binary models.  The
normal model is <code class="reqn">y = \beta_q + A \gamma + Z \delta_q + \epsilon</code> where <em>q</em> is the unknown QTL genotype, <em>A</em>
is a matrix of additive covariates, and <em>Z</em> is a matrix of
covariates that interact with the QTL genotype.  The columns of <em>Z</em>
are forced to be contained in the matrix <em>A</em>.  The binary model is
the logistic regression analog.
</p>
<p>The LOD score is calculated comparing the likelihood of the above
model to that of the null model <code class="reqn">y = \mu + A \gamma + \epsilon</code>.
</p>
<p>Covariates must be numeric matrices.  Individuals with any missing
covariates are discarded.
</p>


<h3>X chromosome</h3>

<p>The X chromosome must be treated specially in QTL mapping.  See Broman
et al. (2006).
</p>
<p>If both males and females are included, male hemizygotes are allowed
to be different from female homozygotes.  Thus, in a backcross, we will
fit separate means for the genotype classes AA, AB, AY, and BY.  In such
cases, sex differences in the phenotype could cause spurious linkage to
the X chromosome, and so the null hypothesis must be changed to allow
for a sex difference in the phenotype.
</p>
<p>Numerous special cases must be considered, as detailed in the following
table.
</p>

<table>
<tr>
<td style="text-align: left;">
<b>BC</b>
 </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> <b>Sexes</b> </td>
<td style="text-align: center;"> <b>Null</b> </td>
<td style="text-align: center;"> <b>Alternative</b> </td>
<td style="text-align: center;"> <b>df</b> </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> both sexes   </td>
<td style="text-align: center;"> sex         </td>
<td style="text-align: center;"> AA/AB/AY/BY        </td>
<td style="text-align: center;"> 2         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> all female   </td>
<td style="text-align: center;"> grand mean  </td>
<td style="text-align: center;"> AA/AB              </td>
<td style="text-align: center;"> 1         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;"> all male     </td>
<td style="text-align: center;"> grand mean  </td>
<td style="text-align: center;"> AY/BY              </td>
<td style="text-align: center;"> 1         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> </td>
<td style="text-align: left;">              </td>
<td style="text-align: center;">             </td>
<td style="text-align: center;">                    </td>
<td style="text-align: center;">           </td>
</tr>
<tr>
<td style="text-align: left;">

<b>F2</b>
 </td>
<td style="text-align: left;"> <b>Direction</b> </td>
<td style="text-align: left;"> <b>Sexes</b>
</td>
<td style="text-align: center;"> <b>Null</b>
</td>
<td style="text-align: center;"> <b>Alternative</b> </td>
<td style="text-align: center;"> <b>df</b> </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> Both             </td>
<td style="text-align: left;"> both sexes  </td>
<td style="text-align: center;"> femaleF/femaleR/male </td>
<td style="text-align: center;"> AA/ABf/ABr/BB/AY/BY </td>
<td style="text-align: center;"> 3         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;">                  </td>
<td style="text-align: left;"> all female  </td>
<td style="text-align: center;"> pgm        </td>
<td style="text-align: center;"> AA/ABf/ABr/BB      </td>
<td style="text-align: center;"> 2         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;">                  </td>
<td style="text-align: left;"> all male    </td>
<td style="text-align: center;"> grand mean </td>
<td style="text-align: center;"> AY/BY              </td>
<td style="text-align: center;"> 1         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> Forward          </td>
<td style="text-align: left;"> both sexes  </td>
<td style="text-align: center;"> sex        </td>
<td style="text-align: center;"> AA/AB/AY/BY        </td>
<td style="text-align: center;"> 2         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;">                  </td>
<td style="text-align: left;"> all female  </td>
<td style="text-align: center;"> grand mean </td>
<td style="text-align: center;"> AA/AB              </td>
<td style="text-align: center;"> 1         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;">                  </td>
<td style="text-align: left;"> all male    </td>
<td style="text-align: center;"> grand mean </td>
<td style="text-align: center;"> AY/BY              </td>
<td style="text-align: center;"> 1         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;"> Backward         </td>
<td style="text-align: left;"> both sexes  </td>
<td style="text-align: center;"> sex        </td>
<td style="text-align: center;"> AB/BB/AY/BY        </td>
<td style="text-align: center;"> 2         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;">                  </td>
<td style="text-align: left;"> all female  </td>
<td style="text-align: center;"> grand mean </td>
<td style="text-align: center;"> AB/BB              </td>
<td style="text-align: center;"> 1         </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
<td style="text-align: left;">                  </td>
<td style="text-align: left;"> all male    </td>
<td style="text-align: center;"> grand mean </td>
<td style="text-align: center;"> AY/BY              </td>
<td style="text-align: center;"> 1         </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>In the case that the number of degrees of freedom for the linkage test
for the X chromosome is different from that for autosomes, a separate
X-chromosome LOD threshold is recommended.  Autosome- and
X-chromosome-specific LOD thresholds may be estimated by permutation
tests with <code>scanone</code> by setting <code>n.perm</code>&gt;0 and using
<code>perm.Xsp=TRUE</code>.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>; Hao Wu</p>


<h3>References</h3>

<p>Boyartchuk,  V. L., Broman, K. W., Mosher, R. E., D'Orazio
S. E. F., Starnbach, M. N. and Dietrich, W. F. (2001) Multigenic
control of <em>Listeria monocytogenes</em> susceptibility in
mice. <em>Nature Genetics</em> <b>27</b>, 259–260.
</p>
<p>Broman,  K. W. (2003) Mapping quantitative trait loci in the case of a
spike in the phenotype distribution. <em>Genetics</em> <b>163</b>,
1169–1175.
</p>
<p>Broman, K. W., Sen, Ś, Owens, S. E., Manichaikul, A.,
Southard-Smith, E. M. and Churchill G. A. (2006)  The X chromosome in
quantitative trait locus mapping.  <em>Genetics</em>, <b>174</b>, 2151–2158.
</p>
<p>Churchill, G. A. and Doerge, R. W. (1994) Empirical threshold values for
quantitative trait mapping.  <em>Genetics</em> <b>138</b>, 963–971.
</p>
<p>Dempster, A. P., Laird, N. M. and Rubin, D. B. (1977) Maximum
likelihood from incomplete data via the EM algorithm.  <em>J. Roy.
Statist. Soc.</em> B, <b>39</b>, 1–38.
</p>
<p>Feenstra, B., Skovgaard, I. M. and Broman, K. W. (2006) Mapping
quantitative trait loci by an extension of the Haley-Knott regression
method using estimating equations. <em>Genetics</em>, <b>173</b>,
2111–2119.
</p>
<p>Haley, C. S. and Knott, S. A. (1992) A simple regression method for mapping
quantitative trait loci in line crosses using flanking markers.
<em>Heredity</em> <b>69</b>, 315–324.
</p>
<p>Kruglyak, L. and Lander, E. S. (1995) A nonparametric approach for
mapping quantitative trait loci.  <em>Genetics</em> <b>139</b>,
1421–1428.
</p>
<p>Lander, E. S. and Botstein, D. (1989) Mapping Mendelian factors underlying
quantitative traits using RFLP linkage maps.  <em>Genetics</em>
<b>121</b>, 185–199.
</p>
<p>Sen, Ś. and Churchill, G. A. (2001) A statistical framework for quantitative
trait mapping.  <em>Genetics</em> <b>159</b>, 371–387.
</p>
<p>Soller, M., Brody, T. and Genizi, A. (1976) On the power of experimental
designs for the detection of linkage between marker loci and
quantitative loci in crosses between inbred lines.
<em>Theor. Appl. Genet.</em> <b>47</b>, 35–39.
</p>
<p>Xu, S., and Atchley, W.R. (1996) Mapping quantitative trait loci for
complex binary diseases using line crosses. <em>Genetics</em>
<b>143</b>, 1417–1424.
</p>


<h3>See Also</h3>

 <p><code>plot.scanone</code>,
<code>summary.scanone</code>, <code>scantwo</code>,
<code>calc.genoprob</code>, <code>sim.geno</code>,
<code>max.scanone</code>,
<code>summary.scanoneperm</code>,
<code>-.scanone</code>, <code>+.scanone</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">###################
# Normal Model
###################
data(hyper)

# Genotype probabilities for EM and H-K
## Not run: hyper &lt;- calc.genoprob(hyper, step=2.5)

out.em &lt;- scanone(hyper, method="em")
out.hk &lt;- scanone(hyper, method="hk")

# Summarize results: peaks above 3
summary(out.em, thr=3)
summary(out.hk, thr=3)

# An alternate method of summarizing:
#     patch them together and then summarize
out &lt;- c(out.em, out.hk)
summary(out, thr=3, format="allpeaks")

# Plot the results
plot(out.hk, out.em)
plot(out.hk, out.em, chr=c(1,4), lty=1, col=c("blue","black"))

# Imputation; first need to run sim.geno
# Do just chromosomes 1 and 4, to save time
## Not run: hyper.c1n4 &lt;- sim.geno(subset(hyper, chr=c(1,4)),
                       step=2.5, n.draws=8)

## End(Not run)
out.imp &lt;- scanone(hyper.c1n4, method="imp")
summary(out.imp, thr=3)

# Plot all three results
plot(out.imp, out.hk, out.em, chr=c(1,4), lty=1,
     col=c("red","blue","black"))

# extended Haley-Knott
out.ehk &lt;- scanone(hyper, method="ehk")
plot(out.hk, out.em, out.ehk, chr=c(1,4))

# Permutation tests
## Not run: permo &lt;- scanone(hyper, method="hk", n.perm=1000)


# Threshold from the permutation test
summary(permo, alpha=c(0.05, 0.10))

# Results above the 0.05 threshold
summary(out.hk, perms=permo, alpha=0.05)


####################
# scan with square-root of phenotype
#   (Note that pheno.col can be a vector of phenotype values)
####################
out.sqrt &lt;- scanone(hyper, pheno.col=sqrt(pull.pheno(hyper, 1)))
plot(out.em - out.sqrt, ylim=c(-0.1,0.1),
     ylab="Difference in LOD")
abline(h=0, lty=2, col="gray")


####################
# Stratified permutations
####################
extremes &lt;- (nmissing(hyper)/totmar(hyper) &lt; 0.5)

## Not run: operm.strat &lt;- scanone(hyper, method="hk", n.perm=1000,
                       perm.strata=extremes)

## End(Not run)

summary(operm.strat)



####################
# X-specific permutations
####################
data(fake.f2)

## Not run: fake.f2 &lt;- calc.genoprob(fake.f2, step=2.5)


# genome scan
out &lt;- scanone(fake.f2, method="hk")

# X-chr-specific permutations
## Not run: operm &lt;- scanone(fake.f2, method="hk", n.perm=1000, perm.Xsp=TRUE)


# thresholds
summary(operm)

# scanone summary with p-values
summary(out, perms=operm, alpha=0.05, pvalues=TRUE)



###################
# Non-parametric
###################
out.np &lt;- scanone(hyper, model="np")
summary(out.np, thr=3)

# Plot with previous results
plot(out.np, chr=c(1,4), lty=1, col="green")
plot(out.imp, out.hk, out.em, chr=c(1,4), lty=1,
     col=c("red","blue","black"), add=TRUE)

###################
# Two-part Model
###################
data(listeria)

## Not run: listeria &lt;- calc.genoprob(listeria,step=2.5)

out.2p &lt;- scanone(listeria, model="2part", upper=TRUE)
summary(out.2p, thr=c(5,3,3), format="allpeaks")

# Plot all three LOD scores together
plot(out.2p, out.2p, out.2p, lodcolumn=c(2,3,1), lty=1, chr=c(1,5,13),
     col=c("red","blue","black"))

# Permutation test
## Not run: permo &lt;- scanone(listeria, model="2part", upper=TRUE,
                 n.perm=1000)

## End(Not run)

# Thresholds
summary(permo)

###################
# Binary model
###################
binphe &lt;- as.numeric(pull.pheno(listeria,1)==264)
out.bin &lt;- scanone(listeria, pheno.col=binphe, model="binary")
summary(out.bin, thr=3)

# Plot LOD for binary model with LOD(p) from 2-part model
plot(out.bin, out.2p, lodcolumn=c(1,2), lty=1, col=c("black", "red"),
     chr=c(1,5,13))

# Permutation test
## Not run: permo &lt;- scanone(listeria, pheno.col=binphe, model="binary",
                 n.perm=1000)

## End(Not run)

# Thresholds
summary(permo)

###################
# Covariates
###################
data(fake.bc)

## Not run: fake.bc &lt;- calc.genoprob(fake.bc, step=2.5)


# genome scans without covariates
out.nocovar &lt;- scanone(fake.bc)

# genome scans with covariates
ac &lt;- pull.pheno(fake.bc, c("sex","age"))
ic &lt;- pull.pheno(fake.bc, "sex")

out.covar &lt;- scanone(fake.bc, pheno.col=1,
                     addcovar=ac, intcovar=ic)
summary(out.nocovar, thr=3)
summary(out.covar, thr=3)
plot(out.covar, out.nocovar, chr=c(2,5,10))
</code></pre>


</div>