<div class="container">

<table style="width: 100%;"><tr>
<td>iplotScanone</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Interactive LOD curve</h2>

<h3>Description</h3>

<p>Creates an interactive graph of a single-QTL genome scan, as
calculated by [qtl::scanone()]. If 'cross' is
provided, the LOD curves are linked to a phenotype x genotype plot
for a marker: Click on a marker on the LOD curve and see the
corresponding phenotype x genotype plot.
</p>


<h3>Usage</h3>

<pre><code class="language-R">iplotScanone(
  scanoneOutput,
  cross = NULL,
  lodcolumn = 1,
  pheno.col = 1,
  chr = NULL,
  pxgtype = c("ci", "raw"),
  fillgenoArgs = NULL,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scanoneOutput</code></td>
<td>
<p>Object of class '"scanone"', as output
from [qtl::scanone()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>(Optional) Object of class '"cross"', see
[qtl::read.cross()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodcolumn</code></td>
<td>
<p>Numeric value indicating LOD score column to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>(Optional) Phenotype column in cross object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>(Optional) Vector indicating the chromosomes
for which LOD scores should be calculated. This should be a vector
of character strings referring to chromosomes by name; numeric
values are converted to strings. Refer to chromosomes with a
preceding - to have all chromosomes but those considered. A logical
(TRUE/FALSE) vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pxgtype</code></td>
<td>
<p>If phenotype x genotype plot is to be shown, should
it be with means <code class="reqn">\pm</code> 2 SE ('"ci"'), or raw
phenotypes ('"raw"')?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fillgenoArgs</code></td>
<td>
<p>List of named arguments to pass to
[qtl::fill.geno()], if needed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chartOpts</code></td>
<td>
<p>A list of options for configuring the chart (see
the coffeescript code). Each element must be named using the
corresponding option.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If 'cross' is provided, [qtl::fill.geno()]
is used to impute missing genotypes. In this case, arguments to
[qtl::fill.geno()] are passed as a list, for example
'fillgenoArgs=list(method="argmax", error.prob=0.002,
map.function="c-f")'.
</p>
<p>With 'pxgtype="raw"', individual IDs (viewable when hovering
over a point in the phenotype-by-genotype plot) are taken from the
input 'cross' object, using the [qtl::getid()]
function in R/qtl.
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotMScanone()], [iplotPXG()], [iplotMap()]
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(qtl)
data(hyper)
hyper &lt;- calc.genoprob(hyper, step=1)
out &lt;- scanone(hyper)

# iplotScanone with no effects
iplotScanone(out, chr=c(1, 4, 6, 7, 15))


# iplotScanone with CIs
iplotScanone(out, hyper, chr=c(1, 4, 6, 7, 15))


# iplotScanone with raw phe x gen
iplotScanone(out, hyper, chr=c(1, 4, 6, 7, 15),
             pxgtype='raw')

</code></pre>


</div>