<div class="container">

<table style="width: 100%;"><tr>
<td>iplotScantwo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Interactive plot of 2d genome scan</h2>

<h3>Description</h3>

<p>Creates an interactive plot of the results of
[qtl::scantwo()], for a two-dimensional, two-QTL genome
scan.
</p>


<h3>Usage</h3>

<pre><code class="language-R">iplotScantwo(
  scantwoOutput,
  cross = NULL,
  lodcolumn = 1,
  pheno.col = 1,
  chr = NULL,
  chartOpts = NULL,
  digits = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scantwoOutput</code></td>
<td>
<p>Output of [qtl::scantwo()]</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>(Optional) Object of class '"cross"', see
[qtl::read.cross()].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lodcolumn</code></td>
<td>
<p>Numeric value indicating LOD score column to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno.col</code></td>
<td>
<p>(Optional) Phenotype column in cross object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>(Optional) Optional vector indicating the chromosomes
for which LOD scores should be calculated. This should be a vector
of character strings referring to chromosomes by name; numeric
values are converted to strings. Refer to chromosomes with a
preceding - to have all chromosomes but those considered. A logical
(TRUE/FALSE) vector may also be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chartOpts</code></td>
<td>
<p>A list of options for configuring the chart.  Each
element must be named using the corresponding option.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digits</code></td>
<td>
<p>Round data to this number of significant digits
before passing to the chart function. (Use NULL to not round.)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The estimated QTL effects, and the genotypes in the
phenotype x genotype plot, in the right-hand panels, are derived
following a single imputation to fill in missing data, and so are a
bit crude.
</p>
<p>Note that the usual 'height' and 'width' options in
'chartOpts' are ignored here. Instead, you may provide
'pixelPerCell' (number of pixels per cell in the heat map),
'chrGap' (gaps between chr in heat map), 'wright' (width
in pixels of right panels), and 'hbot' (height in pixels of
each of the lower panels)
</p>


<h3>Value</h3>

<p>An object of class 'htmlwidget' that will
intelligently print itself into HTML in a variety of contexts
including the R console, within R Markdown documents, and within
Shiny output bindings.
</p>


<h3>See Also</h3>

<p>[iplotScanone()]
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(qtl)
data(fake.f2)

fake.f2 &lt;- calc.genoprob(fake.f2, step=5)
out &lt;- scantwo(fake.f2, method="hk", verbose=FALSE)

iplotScantwo(out, fake.f2)

</code></pre>


</div>