<div class="container">

<table style="width: 100%;"><tr>
<td>mAIC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Multiple QTL AIC
</h2>

<h3>Description</h3>

<p>Multiple QTL model selection by AIC criterion.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mAIC(y, x, gdat, prdat = NULL, vc = NULL, chrIdx, xin, k = 2,
   direction = c("both","backward","forward"), ext = FALSE, msg = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>A numeric vector or a numeric matrix of one column (representing a phenotype for instance).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>A data frame or matrix, representing covariates if not missing.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gdat</code></td>
<td>

<p>Genotype data. Should be a matrix or a data frame, with each row representing an observation and each column a marker locus. The column names should be marker names. Numeric coding of genotype is treated as numeric. Ignored if <code>prdat</code> is an object from <code>genoProb</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vc</code></td>
<td>

<p>An object from <code>estVC</code> or <code>aicVC</code>, or an estimated variance-covariance matrix induced by relatedness. The scan will assume no polygenic variation if <code>vc</code> is NULL.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prdat</code></td>
<td>

<p>An object from <code>genoProb</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrIdx</code></td>
<td>

<p>Chromsome index of markers in columns of <code>gdat</code> if given. Ignored if <code>prdat</code> is an object from <code>genoProb</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xin</code></td>
<td>

<p>Vector indicating whether a locus is already in the model.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>

<p>Penalty on a parameter. The selection criterion is the known "AIC" if <code>k = 2</code> and is "BIC" if <code>k = log(n)</code> where "n" is the sample size.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>direction</code></td>
<td>

<p>The mode of search: "both", "forward" or "backward" with default "both".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ext</code></td>
<td>

<p>A logical variable. True if ones wants more exhaustive search.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>msg</code></td>
<td>

<p>A logical variable. True if ones wants to track the process for monitoring purpose.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Makes use of "Haley-Knott" method (Haley and Knott 1992) if <code>prdat</code> is an object from <code>genoProb</code>.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<p><code>model</code>: the resulting model;
</p>
<p><code>aic</code>: AIC of the model;
</p>
<p><code>snp</code>: selected SNPs.
</p>
<p><code>xin</code>: vector indicating whether a SNP is selected.
</p>


<h3>Note</h3>

<p>Currently only suitable for advanced intercross lines (or diallelic data).
</p>


<h3>References</h3>

<p>Haley, C. S., and S. A. Knott (1992). A simple regression method for mapping quantitative trait loci in line crosses using flanking markers. Heredity 69: 315-324.
</p>


<h3>See Also</h3>

<p><code>optim</code>, <code>genoProb</code> and <code>aicVC</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(miscEx)

## Not run: 
# impute missing genotypes
pheno&lt;- pdatF8[!is.na(pdatF8$bwt) &amp; !is.na(pdatF8$sex),]
ii&lt;- match(rownames(pheno), rownames(gdatF8))
geno&lt;- gdatF8[ii,]
ii&lt;- match(rownames(pheno), rownames(gmF8$AA))
v&lt;- list(A=gmF8$AA[ii,ii], D=gmF8$DD[ii,ii])

gdat.imp&lt;- genoImpute(geno, gmap=gmapF8,
   gr=8, na.str=NA)
# estimate variance components
o&lt;- estVC(y=pheno$bwt, x=pheno$sex, v=v)

# run 'genoProb'
gdtmp&lt;- geno
   gdtmp&lt;- replace(gdtmp,is.na(gdtmp),0)
prDat&lt;- genoProb(gdat=gdtmp, gmap=gmapF8,
   gr=8, method="Haldane", msg=TRUE)

# genome scan
llk.hk&lt;- scanOne(y=pheno$bwt, x=pheno$sex, prdat=prDat, vc=o)
xin&lt;- llk.hk$LRT &gt; 10

# run 'mAIC' based on genome scan results
mg&lt;- mAIC(y=pheno$bwt, x=pheno$sex, prdat=prDat, vc=o, xin=xin,
   k=5, direction="back", msg=TRUE)
mg$model$value # likelihood of the final model

## End(Not run)
</code></pre>


</div>