<div class="container">

<table style="width: 100%;"><tr>
<td>genoProb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Probability of a Genotype.
</h2>

<h3>Description</h3>

<p>Calculate the probability of a genotype at a locus conditional on the genotypes of its flanking markers in advance intercross lines (AIL).
</p>


<h3>Usage</h3>

<pre><code class="language-R">genoProb(gdat, gmap, step, gr = 2, pos = NULL, method=c("Haldane", "Kosambi"),
   msg = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gdat</code></td>
<td>

<p>Genotype data. Should be a matrix or a data frame, with each row representing an observation and each column a marker locus. The column names should be marker names. Each entry should be 1, 2, 3 or 0, corresponding to "AA", "AB", "BB" or missing genotype.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gmap</code></td>
<td>

<p>A genetic map. Should be data frame (snp, chr, dist,...), where "snp" is the SNP (marker) name, "chr" is the chromosome where the "snp" is, and "dist" is the genetic distance in centi-Morgan (cM) from the left of the chromosome.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>step</code></td>
<td>

<p>Optional. If specified, it is the maximum "cumulative" genetic distance (in cM) between two adjacent loci for which the probabilities are calculated. The genetic distance corresponds to the "cumulative" recombination rate at <code>gr</code>-th generation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gr</code></td>
<td>

<p>The generation under consideration.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>

<p>Data frame (chr, dist, snp, ...). If given, <code>step</code> will be ignored.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>Whether "Haldane" or "Kosambi" mapping function should be used.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>msg</code></td>
<td>

<p>A logical variable. If TRUE, certain information will be printed out during calculation.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The "cumulative" genetic distance between any two adjacent loci for which probabilities are calculated is not larger than <code>step</code>. If <code>step</code> is missing or <code>step = Inf</code>, probabilities will only be calculated at loci in both the columns of <code>gdat</code> and the rows of <code>gmap</code>. If <code>step</code> is small, a large set of putative loci will be considered, including all loci defined by the columns of <code>gdat</code> and the rows of <code>gmap</code>. 
</p>


<h3>Value</h3>

<p>Probabilities for genotypes as well as genetic map information (snp,chr,dist)
</p>
<table><tr style="vertical-align: top;">
<td><code>pr</code></td>
<td>
<p>A 3-D array with the first dimension corresponding to that of <code>gdat</code>, the second to three genotype and the third to the putative loci. The probabilities will be -1 if not imputable, which happens when the genotype data is missing at all loci on the chromosome.</p>
</td>
</tr></table>
<h3>Note</h3>

<p>Currently only suitable for advanced intercross lines.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(miscEx)

## Not run: 
# briefly look at genotype data
sum(is.na(gdatF8))
gdatF8[1:5,1:5]

gdtmp&lt;- gdatF8
   gdtmp&lt;- replace(gdtmp,is.na(gdtmp),0)
# In case an individual is not imputable, then
# one needs to assign genotypes manually
prDat&lt;- genoProb(gdat=gdtmp, gmap=gmapF8, gr=8, method="Haldane", msg=TRUE)
prDat$pr[1:5,,1:5]

## End(Not run)
</code></pre>


</div>