<div class="container">

<table style="width: 100%;"><tr>
<td>read_scoring_file</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read a polygenic scoring file</h2>

<h3>Description</h3>

<p>This function imports a PGS scoring file. For more information about the
scoring file schema check <code>vignette("pgs-scoring-file", package =
"quincunx")</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_scoring_file(
  source,
  harmonized = FALSE,
  assembly = c("GRCh38", "GRCh37"),
  protocol = "http",
  metadata_only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>source</code></td>
<td>
<p>PGS scoring file. This can be specified in three forms: (i) a
PGS identifier, e.g. <code>"PGS000001"</code>, (ii) a path to a local file, e.g.
<code>"~/PGS000001.txt"</code> or <code>"~/PGS000001.txt.gz"</code> or (iii) a direct
URL to the PGS Catalog FTP server, e.g.
<code>"http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>harmonized</code></td>
<td>
<p>Whether to read an alternative, harmonized version of the
PGS scoring file. This version contains harmonized variant information.
This information is provided in extra columns whose names are prefixed with
<code>"hm_"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assembly</code></td>
<td>
<p>If <code>harmonized</code> is <code>TRUE</code>, <code>assembly</code> indicates which the
genome assembly to choose for the harmonized variant data. <code>assembly</code> must
be either <code>"GRCh38"</code> (default) or <code>"GRCh37"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protocol</code></td>
<td>
<p>Network protocol for communication with the PGS Catalog FTP
server: either <code>"http"</code> or <code>"ftp"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metadata_only</code></td>
<td>
<p>Whether to read only the comment block (header) from the
scoring file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The returned value is a named list. The names are copied from the
arguments passed in <code>source</code>. Each element of the list contains
another list of two elements: <code>"metadata"</code> and <code>"data"</code>. The
"metadata" element contains data parsed from the header of the PGS scoring
file. The "data" element contains a data frame with as many rows as
variants that constitute the PGS score. The columns can vary. There are
mandatory and optional columns. The mandatory columns are those that
identify the variant, effect allele (<code>effect_allele</code>), and its
respective weight (<code>effect_weight</code>) in the score. The columns that
identify the variant can either be the <code>rsID</code> or the combination of
<code>chr_name</code> and <code>chr_position</code>. The "data" element will be
<code>NULL</code> is argument <code>metadata_only</code> is <code>TRUE</code>. For more
information about the scoring file schema check
<code>vignette("pgs-scoring-file", package = "quincunx")</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Read a PGS scoring file by PGS ID
# (internally, it translates the PGS ID
#  to the corresponding FTP URL)
try(read_scoring_file("PGS000655"))

# Equivalent to `read_scoring_file("PGS000655")`
url &lt;- paste0(
  "http://ftp.ebi.ac.uk/",
  "pub/databases/spot/pgs/scores/",
  "PGS000655/ScoringFiles/",
  "PGS000655.txt.gz"
)
read_scoring_file(url)


# Reading from a local file
try(read_scoring_file("~/PGS000655.txt.gz"))

## End(Not run)
</code></pre>


</div>